<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Testing Pleiotropy in Multiparental Populations • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Testing Pleiotropy in Multiparental Populations">
<meta property="og:description" content="We implement an
    adaptation of Jiang &amp; Zeng's (1995) &lt;https://www.genetics.org/content/140/3/1111&gt; likelihood ratio test for testing
    the null hypothesis of pleiotropy against the alternative hypothesis,
    two separate quantitative trait loci. The test differs from that in Jiang &amp; Zeng (1995) &lt;https://www.genetics.org/content/140/3/1111&gt; 
    and that in Tian et al. (2016) &lt;doi:10.1534/genetics.115.183624&gt; in
    that our test accommodates multiparental populations.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-46757486-5"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-46757486-5');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/d-variate-analysis.html">Multivariate analysis for two or more traits</a>
    </li>
    <li>
      <a href="articles/multivariate-one-QTL-scans.html">Multivariate, one-QTL scans</a>
    </li>
    <li>
      <a href="articles/recla-analysis.html">Recla Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="qtl2pleio" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#qtl2pleio" class="anchor"></a>qtl2pleio</h1></div>

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>qtl2pleio</code> is a software package for use with the <a href="https://cran.r-project.org/">R statistical computing environment</a>. <code>qtl2pleio</code> is freely available for download and use. I share it under the <a href="https://opensource.org/licenses/mit-license.php">MIT license</a>. The user will also want to download and install the <a href="https://kbroman.org/qtl2/"><code>qtl2</code> R package</a>.</p>
<p>Click <a href="https://mybinder.org/v2/gh/fboehm/qtl2pleio/master?urlpath=rstudio">here</a> to explore <code>qtl2pleio</code> within a live <a href="https://rstudio.com/">Rstudio</a> session in “the cloud”.</p>
</div>
<div id="contributor-guidelines" class="section level2">
<h2 class="hasAnchor">
<a href="#contributor-guidelines" class="anchor"></a>Contributor guidelines</h2>
<p>We eagerly welcome contributions to <code>qtl2pleio</code>. All pull requests will be considered. Features requests and bug reports may be filed as Github issues. All contributors must abide by the <a href="https://github.com/fboehm/qtl2pleio/blob/master/CONDUCT.md">code of conduct</a>.</p>
</div>
<div id="technical-support" class="section level2">
<h2 class="hasAnchor">
<a href="#technical-support" class="anchor"></a>Technical support</h2>
<p>For technical support, please open a Github issue. If you’re just getting started with <code>qtl2pleio</code>, please examine the <a href="https://fboehm.us/software/qtl2pleio">vignettes</a> on the <a href="https://fboehm.us/software/qtl2pleio">package’s web site</a>. You can also email <a href="mailto:frederick.boehm@gmail.com" class="email">frederick.boehm@gmail.com</a> for assistance.</p>
</div>
<div id="goals" class="section level2">
<h2 class="hasAnchor">
<a href="#goals" class="anchor"></a>Goals</h2>
<p>The goal of <code>qtl2pleio</code> is, for a pair of traits that show evidence for a QTL in a common region, to distinguish between pleiotropy (the null hypothesis, that they are affected by a common QTL) and the alternative that they are affected by separate QTL. It extends the likelihood ratio test of <a href="https://www.genetics.org/content/140/3/1111.long">Jiang and Zeng (1995)</a> for multiparental populations, such as Diversity Outbred mice, including the use of multivariate polygenic random effects to account for population structure. <code>qtl2pleio</code> data structures are those used in the <a href="https://kbroman.org/qtl2/"><code>rqtl/qtl2</code></a> package.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install qtl2pleio, use <code>install_github()</code> from the <a href="https://devtools.r-lib.org">devtools</a> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"qtl2pleio"</span><span class="op">)</span></code></pre></div>
<p>You may also wish to install the <a href="https://kbroman.org/qtl2/">R/qtl2</a>. We will use it below.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"qtl2"</span><span class="op">)</span></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>Below, we walk through an example analysis with Diversity Outbred mouse data. We need a number of preliminary steps before we can perform our test of pleiotropy vs. separate QTL. Many procedures rely on the R package <code>qtl2</code>. We first load the <code>qtl2</code> and <code>qtl2pleio</code> packages.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://kbroman.org/qtl2">qtl2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fboehm/qtl2pleio">qtl2pleio</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
<div id="reading-data-from-qtl2data-repository-on-github" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-data-from-qtl2data-repository-on-github" class="anchor"></a>Reading data from <code>qtl2data</code> repository on github</h3>
<p>We’ll consider the <a href="https://github.com/rqtl/qtl2data/tree/master/DOex/"><code>DOex</code></a> data in the <a href="https://github.com/rqtl/qtl2data/"><code>qtl2data</code></a> repository. We’ll download the DOex.zip file before calculating founder allele dosages.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,
               <span class="st">"qtl2data/master/DOex/DOex.zip"</span><span class="op">)</span>
<span class="va">DOex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/read_cross2.html">read_cross2</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/calc_genoprob.html">calc_genoprob</a></span><span class="op">(</span><span class="va">DOex</span><span class="op">)</span></code></pre></div>
<p>Let’s calculate the founder allele dosages from the 36-state genotype probabilities.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/genoprob_to_alleleprob.html">genoprob_to_alleleprob</a></span><span class="op">(</span><span class="va">probs</span><span class="op">)</span></code></pre></div>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pr</span><span class="op">)</span>
<span class="co">#&gt; [1] "2" "3" "X"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">pr</span><span class="op">$</span><span class="va">`2`</span><span class="op">)</span>
<span class="co">#&gt; [1] 261   8 127</span></code></pre></div>
<p>We see that <code>pr</code> is a list of 3 three-dimensional arrays - one array for each of 3 chromosomes.</p>
</div>
<div id="kinship-calculations" class="section level3">
<h3 class="hasAnchor">
<a href="#kinship-calculations" class="anchor"></a>Kinship calculations</h3>
<p>For our statistical model, we need a kinship matrix. We get one with the <code>calc_kinship</code> function in the <code>rqtl/qtl2</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kinship</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/calc_kinship.html">calc_kinship</a></span><span class="op">(</span>probs <span class="op">=</span> <span class="va">pr</span>, type <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></code></pre></div>
</div>
<div id="statistical-model-specification" class="section level3">
<h3 class="hasAnchor">
<a href="#statistical-model-specification" class="anchor"></a>Statistical model specification</h3>
<p>We use the multivariate linear mixed effects model:</p>
<p><em>v<strong>e</strong>c</em>(<em>Y</em>) = *X<strong>v</strong>e<strong>c<em>(</em>B<em>) + </em>v</strong>e<strong>c<em>(</em>G<em>) + </em>v</strong>e**c<em>(</em>E*)</p>
<p>where <em>Y</em> contains phenotypes, X contains founder allele probabilities and covariates, and B contains founder allele effects. G is the polygenic random effects, while E is the random errors. We provide more details in the vignette.</p>
</div>
<div id="simulating-phenotypes-with-qtl2pleiosim1" class="section level3">
<h3 class="hasAnchor">
<a href="#simulating-phenotypes-with-qtl2pleiosim1" class="anchor"></a>Simulating phenotypes with <code>qtl2pleio::sim1</code>
</h3>
<p>The function to simulate phenotypes in <code>qtl2pleio</code> is <code>sim1</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set up the design matrix, X</span>
<span class="va">pp</span> <span class="op">&lt;-</span> <span class="va">pr</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="co">#we'll work with Chr 3's genotype data</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Next, we prepare a design matrix X</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu">gemma2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gemma2/man/stagger_mats.html">stagger_mats</a></span><span class="op">(</span><span class="va">pp</span><span class="op">[</span> , , <span class="fl">50</span><span class="op">]</span>, <span class="va">pp</span><span class="op">[</span> , , <span class="fl">50</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># assemble B matrix of allele effects</span>
<span class="va">B</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">8</span>, ncol <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># set.seed to ensure reproducibility</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2018</span><span class="op">-</span><span class="fl">01</span><span class="op">-</span><span class="fl">30</span><span class="op">)</span>
<span class="va">Sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calc_Sigma.html">calc_Sigma</a></span><span class="op">(</span>Vg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, Ve <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, kinship <span class="op">=</span> <span class="va">kinship</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co"># call to sim1</span>
<span class="va">Ypre</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/sim1.html">sim1</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">X</span>, B <span class="op">=</span> <span class="va">B</span>, Sigma <span class="op">=</span> <span class="va">Sig</span><span class="op">)</span>
<span class="va">Y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="va">Ypre</span>, nrow <span class="op">=</span> <span class="fl">261</span>, ncol <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tr1"</span>, <span class="st">"tr2"</span><span class="op">)</span></code></pre></div>
<p>Let’s perform univariate QTL mapping for each of the two traits in the Y matrix.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/scan1.html">scan1</a></span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>, pheno <span class="op">=</span> <span class="va">Y</span>, kinship <span class="op">=</span> <span class="va">kinship</span><span class="op">)</span></code></pre></div>
<p>Here is a plot of the results.</p>
<p><img src="https://raw.githubusercontent.com/fboehm/qtl2pleio/master/man/figures/README-1d-lod-plots-1.png" width="100%"></p>
<p>And here are the observed QTL peaks with LOD &gt; 8.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/find_peaks.html">find_peaks</a></span><span class="op">(</span><span class="va">s1</span>, map <span class="op">=</span> <span class="va">DOex</span><span class="op">$</span><span class="va">pmap</span>, threshold<span class="op">=</span><span class="fl">8</span><span class="op">)</span>
<span class="co">#&gt;   lodindex lodcolumn chr      pos       lod</span>
<span class="co">#&gt; 1        1       tr1   3 82.77806 20.703383</span>
<span class="co">#&gt; 2        2       tr2   3 82.77806 14.417924</span>
<span class="co">#&gt; 3        2       tr2   X 48.10040  8.231551</span></code></pre></div>
</div>
<div id="perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="section level3">
<h3 class="hasAnchor">
<a href="#perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="anchor"></a>Perform two-dimensional scan as first step in pleiotropy vs. separate QTL hypothesis test</h3>
<p>We now have the inputs that we need to do a pleiotropy vs. separate QTL test. We have the founder allele dosages for one chromosome, <em>i.e.</em>, Chr 3, in the R object <code>pp</code>, the matrix of two trait measurements in <code>Y</code>, and a LOCO-derived kinship matrix, <code>kinship[[2]]</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="reference/scan_pvl.html">scan_pvl</a></span><span class="op">(</span>probs <span class="op">=</span> <span class="va">pp</span>,
                pheno <span class="op">=</span> <span class="va">Y</span>,
                kinship <span class="op">=</span> <span class="va">kinship</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="co"># 2nd entry in kinship list is Chr 3</span>
                start_snp <span class="op">=</span> <span class="fl">38</span>,
                n_snp <span class="op">=</span> <span class="fl">25</span>
                <span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="section level3">
<h3 class="hasAnchor">
<a href="#create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="anchor"></a>Create a profile LOD plot to visualize results of two-dimensional scan</h3>
<p>To visualize results from our two-dimensional scan, we calculate profile LOD for each trait. The code below makes use of the R package <code>ggplot2</code> to plot profile LODs over the scan region.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="va">out</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/calc_profile_lods.html">calc_profile_lods</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/add_pmap.html">add_pmap</a></span><span class="op">(</span>pmap <span class="op">=</span> <span class="va">DOex</span><span class="op">$</span><span class="va">pmap</span><span class="op">$</span><span class="va">`3`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">marker_position</span>, y <span class="op">=</span> <span class="va">profile_lod</span>, colour <span class="op">=</span> <span class="va">trait</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-profile-plot-1.png" width="100%"></p>
</div>
<div id="calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="anchor"></a>Calculate the likelihood ratio test statistic for pleiotropy v separate QTL</h3>
<p>We use the function <code>calc_lrt_tib</code> to calculate the likelihood ratio test statistic value for the specified traits and specified genomic region.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">lrt</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calc_lrt_tib.html">calc_lrt_tib</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0</span></code></pre></div>
</div>
<div id="bootstrap-analysis-to-get-p-values" class="section level3">
<h3 class="hasAnchor">
<a href="#bootstrap-analysis-to-get-p-values" class="anchor"></a>Bootstrap analysis to get p-values</h3>
<p>Before we call <code>boot_pvl</code>, we need to identify the index (on the chromosome under study) of the marker that maximizes the likelihood under the pleiotropy constraint. To do this, we use the <code>qtl2pleio</code> function <code>find_pleio_peak_tib</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">pleio_index</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span><span class="op">(</span><span class="va">out</span>, start_snp <span class="op">=</span> <span class="fl">38</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; log10lik13 </span>
<span class="co">#&gt;         50</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2018</span><span class="op">-</span><span class="fl">11</span><span class="op">-</span><span class="fl">25</span><span class="op">)</span> <span class="co"># set for reproducibility purposes.</span>
<span class="va">b_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="reference/boot_pvl.html">boot_pvl</a></span><span class="op">(</span>probs <span class="op">=</span> <span class="va">pp</span>,
         pheno <span class="op">=</span> <span class="va">Y</span>,
         pleio_peak_index <span class="op">=</span> <span class="va">pleio_index</span>,
         kinship <span class="op">=</span> <span class="va">kinship</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="co"># 2nd element in kinship list is Chr 3</span>
         nboot <span class="op">=</span> <span class="fl">10</span>,
         start_snp <span class="op">=</span> <span class="fl">38</span>,
         n_snp <span class="op">=</span> <span class="fl">25</span>
         <span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">b_out</span> <span class="op">&gt;=</span> <span class="va">lrt</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
</div>
</div>
<div id="citation-information" class="section level2">
<h2 class="hasAnchor">
<a href="#citation-information" class="anchor"></a>Citation information</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"qtl2pleio"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite qtl2pleio in publications use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Boehm FJ, Chesler EJ, Yandell BS, Broman KW (2019) Testing pleiotropy</span>
<span class="co">#&gt;   vs. separate QTL in multiparental populations G3</span>
<span class="co">#&gt;   https://www.g3journal.org/content/9/7/2317</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{Boehm2019testing,</span>
<span class="co">#&gt;     title = {Testing pleiotropy vs. separate QTL in multiparental populations},</span>
<span class="co">#&gt;     author = {Frederick J. Boehm and Elissa J. Chesler and Brian S. Yandell and Karl W. Broman},</span>
<span class="co">#&gt;     journal = {G3},</span>
<span class="co">#&gt;     year = {2019},</span>
<span class="co">#&gt;     volume = {9},</span>
<span class="co">#&gt;     issue = {7},</span>
<span class="co">#&gt;     url = {https://www.g3journal.org/content/9/7/2317},</span>
<span class="co">#&gt;     eprint = {https://www.g3journal.org/content/ggg/9/7/2317.full.pdf},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=qtl2pleio">https://​cloud.r-project.org/​package=qtl2pleio</a>
</li>
<li>Browse source code at <br><a href="https://github.com/fboehm/qtl2pleio/">https://​github.com/​fboehm/​qtl2pleio/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/fboehm/qtl2pleio/issues">https://​github.com/​fboehm/​qtl2pleio/​issues</a>
</li>
<li>Frederick Boehm's Website at <br><a href="https://fboehm.us">https://​fboehm.us</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing qtl2pleio</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Frederick J Boehm <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1644-5931" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/fboehm/qtl2pleio/actions"><img src="https://github.com/fboehm/qtl2pleio/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://cran.r-project.org/package=qtl2pleio"><img src="https://www.r-pkg.org/badges/version/qtl2pleio" alt="CRAN_Status_Badge"></a></li>
<li><a href="https://codecov.io/github/fboehm/qtl2pleio?branch=master"><img src="https://img.shields.io/codecov/c/github/fboehm/qtl2pleio/master.svg" alt="Coverage Status"></a></li>
<li><a href="https://www.repostatus.org/#active"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a></li>
<li><a href="https://joss.theoj.org/papers/10.21105/joss.01435"><img src="https://joss.theoj.org/papers/66bca5dc3d2e72b6259159bad07aafaf/status.svg" alt="status"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/104493705"><img src="https://zenodo.org/badge/104493705.svg" alt="DOI"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Frederick J Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
