

<!DOCTYPE html>
<html>


<head>

  <link href="https://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>qtl2pleio on CRAN</title>
  <link rel="canonical" href="/blog/qtl2pleio-on-cran/" />

  
  <link rel="stylesheet" href="/assets/css/kube.css" />
  <link rel="stylesheet" href="/assets/css/custom.css" />
  <link rel="stylesheet" href="/assets/css/tufte.css" />

  
  <link href='https://fonts.googleapis.com/css?family=Alegreya:400italic,400,700,700italic&subset=latin,latin-ext' rel='stylesheet' type='text/css'>
  <link href='https://fonts.googleapis.com/css?family=Alegreya+SC:400&subset=latin,latin-ext' rel='stylesheet' type='text/css'>
  <link href="https://fonts.googleapis.com/css?family=IM+Fell+English:400,400i&display=swap" rel="stylesheet">
  <link href="https://fonts.googleapis.com/css?family=IM+Fell+English+SC&display=swap" rel="stylesheet">
  <link href="/assets/fonts/fell-flowers/style.css" rel="stylesheet" type="text/css" />
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hack-font@3/build/web/hack-subset.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.0/css/academicons.min.css">
  <link href="/assets/fonts/fontawesome/css/all.css" rel="stylesheet">
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  
  <link rel="apple-touch-icon" sizes="180x180" href="/assets/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
  <link rel="shortcut icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
  <link rel="shortcut icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
  <link rel="manifest" href="/assets/site.webmanifest">
  <link rel="mask-icon" href="/assets/safari-pinned-tab.svg" color="#5bbad5">
  <meta name="msapplication-TileColor" content="#2b5797">
  <meta name="theme-color" content="#ffffff">

  
  

  
  <meta name="description" content="CRAN now hosts the packages qtl2pleio and gemma2. qtl2pleio performs a d-variate, d-QTL scan over a select genomic region. gemma2 is used by qtl2pleio for the inference of multivariate variance components.
They can be installed with:
install.packages(&quot;qtl2pleio&quot;) The statistical model that qtl2pleio fits for each d-tuple of markers (or pseudomarkers) is
\[ vec(Y) = Xvec(B) &#43; vec(G) &#43; vec(E) \] where \(Y\) is a n by d matrix of d traits (for each of n subjects), X is a dn by df block-diagonal matrix of founder allele probabilities, B is a f by d matrix of allele effects for each of d traits, G is a n by d matrix of polygenic random effects, and E is a n by d matrix of random errors.
" />

  <meta property="og:title" content="qtl2pleio on CRAN" />
  <meta property="og:locale" content="en_US" />
  <meta property="og:site_name" content="Frederick Boehm's website" />
  <meta property="og:description" content="  CRAN now hosts the packages qtl2pleio and gemma2. qtl2pleio performs a d-variate, d-QTL scan over a select genomic region. gemma2 is used by qtl2pleio for the inference of multivariate variance components.
They can be installed with:
install.packages(&quot;qtl2pleio&quot;) The statistical model that qtl2pleio fits for each d-tuple of markers (or pseudomarkers) is
\[ vec(Y) = Xvec(B) &#43; vec(G) &#43; vec(E) \] where \(Y\) is a n by d matrix of d traits (for each of n subjects), X is a dn by df block-diagonal matrix of founder allele probabilities, B is a f by d matrix of allele effects for each of d traits, G is a n by d matrix of polygenic random effects, and E is a n by d matrix of random errors." />
  <meta property="og:url" content="/blog/qtl2pleio-on-cran/" />
  <script type="application/ld+json">
    {"@type":"WebSite","url":"\/blog\/qtl2pleio-on-cran\/","headline":"qtl2pleio on CRAN", "name":"Frederick Boehm's website","description":"CRAN now hosts the packages qtl2pleio and gemma2. qtl2pleio performs a d-variate, d-QTL scan over a select genomic region. gemma2 is used by qtl2pleio for the inference of multivariate variance components.\nThey can be installed with:\ninstall.packages(\x26quot;qtl2pleio\x26quot;) The statistical model that qtl2pleio fits for each d-tuple of markers (or pseudomarkers) is\n\\[ vec(Y) = Xvec(B) \x2b vec(G) \x2b vec(E) \\] where \\(Y\\) is a n by d matrix of d traits (for each of n subjects), X is a dn by df block-diagonal matrix of founder allele probabilities, B is a f by d matrix of allele effects for each of d traits, G is a n by d matrix of polygenic random effects, and E is a n by d matrix of random errors.","@context":"http://schema.org"}
  </script>

  
  
</head>

<body>

<div id="topbar" class="is-hidden-print">
	<div class="wrapper">
		<header>
			<div class="is-navbar-container">
				<nav>
				<div class="is-brand">
					<a class="logo" href="/"><img src = "/assets/favicon.ico"></a>
					<a href="#"
						 class="is-push-right-mobile is-shown-mobile icon-kube-menu"
						 data-kube="toggle"
						 data-target="#navbar-main"></a>
				</div>
				</nav>
				<div id="navbar-main" class="is-navbar is-hidden-mobile is-center-mobile">
					<nav>
						<ul>
							<li><a href="/blog">Blog</a></li>
						
							<li><a href="/vitae">Vitae</a></li>
							<li><a href="/software/qtl2pleio">Software</a></li>
							<li><a href="/notebook/_book">Notebook</a></li>
							<li><a href="#connect">Connect</a></li>
						</ul>
					</nav>
					<nav class="is-push-right is-left-mobile" class="is-hidden-mobile">
						



<form method="get" action="https://duckduckgo.com/">
      <input type="search" name="q" maxlength="255" placeholder="Search my site"/>
      <input type="hidden" name="sites" value= "https://fboehm.us/"/>
      <input type="hidden" name="ka" value="Alegreya" />
      <input type="hidden" name="k7" value="#faf8f8" />
      <input type="hidden" name="kj" value="#b33" />
      <input type="hidden" name="ky" value="#fafafa" />
      <input type="hidden" name="kx" value="b" />
      <input type="hidden" name="ko" value="-1" />
      <input type="hidden" name="k1" value="-1" />
      <input type="hidden" name="kt" value="Alegreya Sans" />
</form>

					</nav>
				</div>
			</div>
		</header>
	</div>
</div>


<div class="wrapper">
  <div class="is-row is-stack-20">
    <div class="is-col is-10">&nbsp;</div>
    <div class="is-col is-60">
    <h1 class="is-text-center"><a href="">qtl2pleio on CRAN</a></h1>
    <h4 class="is-bigger is-text-center">January 10, 2020</h4>
    </div>
    <div class="is-col is-30">&nbsp;</div>
  </div>
  <div class="is-row is-stack-20">
  </div>
  <div class="is-row is-stack-20">
    <div class="is-col is-10 is-stack-20">
    </div>
    <div class="is-col is-60">
      <div class="story">
      


<p>CRAN now hosts the packages <code>qtl2pleio</code> and <code>gemma2</code>. <code>qtl2pleio</code> performs a d-variate, d-QTL scan over a select genomic region. <code>gemma2</code> is used by <code>qtl2pleio</code> for the inference of multivariate variance components.</p>
<p>They can be installed with:</p>
<pre class="r"><code>install.packages(&quot;qtl2pleio&quot;)</code></pre>
<p>The statistical model that <code>qtl2pleio</code> fits for each d-tuple of markers (or pseudomarkers) is</p>
<p><span class="math display">\[
vec(Y) = Xvec(B) + vec(G) + vec(E)
\]</span>
where <span class="math inline">\(Y\)</span> is a n by d matrix of d traits (for each of n subjects), X is a dn by df block-diagonal matrix of founder allele probabilities, B is a f by d matrix of allele effects for each of d traits, G is a n by d matrix of polygenic random effects, and E is a n by d matrix of random errors. We assume that</p>
<p><span class="math display">\[
G \sim N(0, K, V_g)
\]</span></p>
<p>and</p>
<p><span class="math display">\[
E \sim N(0, I, V_e)
\]</span></p>
<p>We further assume that <span class="math inline">\(G\)</span> and <span class="math inline">\(E\)</span> are independent. <span class="math inline">\(K\)</span> is an estimated kinship matrix.</p>
<p>In a <a href="https://www.g3journal.org/content/9/7/2317">recent G3 article</a>, we developed
our test of pleiotropy vs. separate QTL and applied it to two behavioral traits
measured on a cohort of Diversity Outbred mice.</p>
<p><code>qtl2pleio</code> “plays nicely” with <a href="https://kbroman.org/qtl2/"><code>qtl2</code></a> and other R packages from the <code>qtl2</code> ecosystem by using the same structures for genotype data, phenotype data, and covariates.</p>
<p>We’re currently investigating our pleiotropy test in the context of expression trait hotspots.</p>
<p>(Last modified: 2020-01-12 13:51:26)</p>

      </div>
      <hr>
      
      <a href="/blog/improvboston-101-student-showcase-video/"><span class="pointer">2</span>&nbsp;ImprovBoston Student Showcase video</a> |
      
      <a href="/blog">All posts</a>
      
      | <a href="/blog/presenting-a-poster-at-tagc2020/">Presenting a poster at TAGC2020&nbsp;<span class="pointer">1</span></a>
      
    </div>
    <div class="is-col is-30"></div>

  </div>
</div>





<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-46757486-5', 'auto');
	
	ga('send', 'pageview');
}
</script>



<footer id="footer" class="is-hidden-print">
  <div class="wrapper">
    <div class="is-row .is-small is-muted">
      <div class="is-col is-33">
        <h3>About</h3>
        <em>
          <p>This is my website. Nothing here is endorsed by my employers, funders, or any organizations I am part of.</p>
          <p>Content on this site is provided under a
            <a href="https://creativecommons.org/licenses/by/4.0/legalcode">Creative Commons (CC-BY) 4.0 license.</a>
            You may re-use this content as long as you indicate Noam Ross's authorship and provide a link back to the original material.
            Source code of the site is provided under the <a href="https://opensource.org/licenses/MIT">MIT license</a>
            and may be reused without restriction.</p>
          <p>This site is gratefully built using a variety of open-source resources and inspirations. Much of the hard work was done by Noam Ross when he built his site. See <a href="https://www.noamross.net/2019/08/09/a-new-website/">here</a> for more about his website and <a href="https://github.com/noamross/noamross.net">here</a> for his source code.</p>
          <p>I've attempted to create a Hugo theme from Noam Ross's materials. The theme is on Github <a href = "https://github.com/fboehm/noamross">here</a>. Please feel free to use and contribute to it.</p>
          <p>I use the Vitruvian Mouse image with permission from <a href="http://daviddeen.com/">David Deen</a>.
</p>
        </em>
      </div>
      <div class="is-col is-33">
        <h3 id="connect">Connect</h3>
        <ul class="fa-ul">
          <li><i class="fa-li fas fa-briefcase fa-fw"></i>I work at <a href="https://www.umassmed.edu/">University of Massachusetts Medical School</a>, Worcester, Massachusetts, USA.</li>
          <li><i class="fa-li fas fa-envelope fa-fw"></i>Email me at <a href="mailto:frederick.boehm@gmail.com">frederick.boehm@gmail.com</a>.</li>
          <li><a href="https://orcid.org/0000-0002-1644-5931"><i class="fa-li ai ai-orcid"></i></a>Track my scholarly work at <a href="https://orcid.org/0000-0002-1644-5931">ORCiD:0000-0002-1644-5931</a>.</li>
          <li><a href="https://scholar.google.com/citations?user=k7PBnv4AAAAJ&hl=en"><i class="fa-li ai ai-google-scholar"></i></a>Or on <a href="https://scholar.google.com/citations?user=k7PBnv4AAAAJ&hl=en">Google Scholar</a>.</li>
          <li><a href="https://github.com/fboehm"><i class="fa-li fab fa-github fa-fw"></i></a>See my open-source code projects at <a href="https://github.com/fboehm">github.com/fboehm</a>.
          <li><a href="/index.xml"><i class="fa-li fas fa-rss fa-fw"></i></a>Follow new posts on this site via the <a href="/index.xml">RSS Feed</a></li>.
        </ul>
      </div>
      <div class="is-col is-33">
        <h3 id="connect">Support</h3>
        <em><p>Some of my work is funded.  In lieu of supporting my volunteer or open-source work, please consider donating to these organizations that I work with or are close to my heart.</p></em>
        <ul class="fa-ul">
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://ropensci.org/donate/">rOpenSci</a> builds tools and community for open science.</li>
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://www.villagehealthworks.org/donate-2">Village Health Works</a> works for health and dignity in Burundi.</li>
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://www.pih.org/pages/donate/give-today">Partners in Health</a> provides health services in many poor communities</li>
        </ul>
      </div>
    </div>
  </div>
</footer>

<script type="text/javascript" src="/assets/js/kube.js"></script>
<script>
  $K.dom(window).on('load', $K.init);
</script>


</body>
</html>
