

<!DOCTYPE html>
<html>


<head>
	<meta name="generator" content="Hugo 0.62.2" />

  <link href="https://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title></title>
  <link rel="canonical" href="/" />

  
  <link rel="stylesheet" href="/assets/css/kube.css" />
  <link rel="stylesheet" href="/assets/css/custom.css" />
  <link rel="stylesheet" href="/assets/css/tufte.css" />

  
  <link href='https://fonts.googleapis.com/css?family=Alegreya:400italic,400,700,700italic&subset=latin,latin-ext' rel='stylesheet' type='text/css'>
  <link href='https://fonts.googleapis.com/css?family=Alegreya+SC:400&subset=latin,latin-ext' rel='stylesheet' type='text/css'>
  <link href="https://fonts.googleapis.com/css?family=IM+Fell+English:400,400i&display=swap" rel="stylesheet">
  <link href="https://fonts.googleapis.com/css?family=IM+Fell+English+SC&display=swap" rel="stylesheet">
  <link href="/assets/fonts/fell-flowers/style.css" rel="stylesheet" type="text/css" />
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hack-font@3/build/web/hack-subset.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.0/css/academicons.min.css">
  <link href="/assets/fonts/fontawesome/css/all.css" rel="stylesheet">
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  
  <link rel="apple-touch-icon" sizes="180x180" href="/assets/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
  <link rel="shortcut icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
  <link rel="shortcut icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
  <link rel="manifest" href="/assets/site.webmanifest">
  <link rel="mask-icon" href="/assets/safari-pinned-tab.svg" color="#5bbad5">
  <meta name="msapplication-TileColor" content="#2b5797">
  <meta name="theme-color" content="#ffffff">

  
  <link rel="alternate" type="application/rss+xml" href="/index.xml" title="fboehm.us" />
  

  
  <meta name="description" content="Welcome to my website! I&#39;m a postdoctoral researcher and biostatistician at the University of Massachusetts Medical School. My research is in statistical methods for systems genetics. My colleagues and I use mice to study biology of complex traits, like body weight, that are affected by many genes and other factors. We design and perform experiments to map quantitative trait loci (QTL). In other words, we identify where in the genome are genes that affect a given complex trait.
" />

  <meta property="og:title" content="" />
  <meta property="og:locale" content="en_US" />
  <meta property="og:site_name" content="Frederick Boehm's website" />
  <meta property="og:description" content="  Welcome to my website! I&#39;m a postdoctoral researcher and biostatistician at the University of Massachusetts Medical School. My research is in statistical methods for systems genetics. My colleagues and I use mice to study biology of complex traits, like body weight, that are affected by many genes and other factors. We design and perform experiments to map quantitative trait loci (QTL). In other words, we identify where in the genome are genes that affect a given complex trait." />
  <meta property="og:url" content="/" />
  <script type="application/ld+json">
    {"@type":"WebSite","url":"\/","headline":"", "name":"Frederick Boehm's website","description":"Welcome to my website! I\x27m a postdoctoral researcher and biostatistician at the University of Massachusetts Medical School. My research is in statistical methods for systems genetics. My colleagues and I use mice to study biology of complex traits, like body weight, that are affected by many genes and other factors. We design and perform experiments to map quantitative trait loci (QTL). In other words, we identify where in the genome are genes that affect a given complex trait.","@context":"http://schema.org"}
  </script>

  
  
</head>

<body>
  <div id="topbar" class="is-hidden-print">
	<div class="wrapper">
		<header>
			<div class="is-navbar-container">
				<nav>
				<div class="is-brand">
					<a class="logo" href="/"><img src = "/assets/favicon.ico"></a>
					<a href="#"
						 class="is-push-right-mobile is-shown-mobile icon-kube-menu"
						 data-kube="toggle"
						 data-target="#navbar-main"></a>
				</div>
				</nav>
				<div id="navbar-main" class="is-navbar is-hidden-mobile is-center-mobile">
					<nav>
						<ul>
							<li><a href="/blog">Blog</a></li>
						
							<li><a href="/vitae">Vitae</a></li>
							<li><a href="/notebook/_book">Notebook</a></li>
							<li><a href="#connect">Connect</a></li>
						</ul>
					</nav>
					<nav class="is-push-right is-left-mobile" class="is-hidden-mobile">
						



<form method="get" action="https://duckduckgo.com/">
      <input type="search" name="q" maxlength="255" placeholder="Search my site"/>
      <input type="hidden" name="sites" value= "https://fboehm.us/"/>
      <input type="hidden" name="ka" value="Alegreya" />
      <input type="hidden" name="k7" value="#faf8f8" />
      <input type="hidden" name="kj" value="#b33" />
      <input type="hidden" name="ky" value="#fafafa" />
      <input type="hidden" name="kx" value="b" />
      <input type="hidden" name="ko" value="-1" />
      <input type="hidden" name="k1" value="-1" />
      <input type="hidden" name="kt" value="Alegreya Sans" />
</form>

					</nav>
				</div>
			</div>
		</header>
	</div>
</div>

  <div class="wrapper">
    <div class="is-row is-stack-40">
      <div class="is-col is-100 story is-big">
        <img class="header-img is-stack-20" src="/assets/img/header-pic.jpg">
        <p>Welcome to my website!  I'm a postdoctoral researcher and biostatistician at the University of Massachusetts Medical School. My research is in statistical methods for systems genetics. My colleagues and I use mice to study biology of complex traits, like body weight, that are affected by many genes and other factors. We design and perform experiments to map quantitative trait loci (QTL). In other words, we identify where in the genome are genes that affect a given complex trait.</p>

      </div>
    </div>
      <div class="is-row">
        <div class="is-col is-66 is-stack-20">
          <div class="is-row is-stack-20">
            <h2 class="is-center">Blog</h2>
          </div>

          
          <div class="is-row is-stack-20">
            <div class="is-col is-33 dateblock">
              <h3><a href="/blog/presenting-a-poster-at-tagc2020/">Presenting a poster at TAGC2020</a></h3>
              <h4>27 April 2020</h4>
            </div>
            <div class="is-col is-66 summary">
              
              


<p>I have the opportunity to present a <a href="https://tagc2020.figshare.com/articles/Enhancing_QTL_detection_power_in_multiparental_populations/12142350">poster</a> at The Allied Genetics
Conference (TAGC) 2020. I’ve posted both a pdf of the poster and a
mp4 narrated video tour of the poster
<a href="https://tagc2020.figshare.com/articles/Enhancing_QTL_detection_power_in_multiparental_populations/12142350">here</a>.</p>
<p>I’d love to address your questions and hear your suggestions in the Q &amp; A session on <strong>Thursday, April 30, at 1:30pm Eastern Time</strong>.
My poster number is <strong>1333A</strong>.</p>
<p>(Last modified: lubridate::now())</p>

              
            </div>
          </div>
          
          <div class="is-row is-stack-20">
            <div class="is-col is-33 dateblock">
              <h3><a href="/blog/qtl2pleio-on-cran/">qtl2pleio on CRAN</a></h3>
              <h4>10 January 2020</h4>
            </div>
            <div class="is-col is-66 summary">
              
              CRAN now hosts the packages qtl2pleio and gemma2. qtl2pleio performs a d-variate, d-QTL scan over a select genomic region. gemma2 is used by qtl2pleio for the inference of multivariate variance components.
They can be installed with:
install.packages(&quot;qtl2pleio&quot;) The statistical model that qtl2pleio fits for each d-tuple of markers (or pseudomarkers) is
\[ vec(Y) = Xvec(B) + vec(G) + vec(E) \] where \(Y\) is a n by d matrix of d traits (for each of n subjects), X is a dn by df block-diagonal matrix of founder allele probabilities, B is a f by d matrix of allele effects for each of d traits, G is a n by d matrix of polygenic random effects, and E is a n by d matrix of random errors.
              <p class="is-text-right"><a href="/blog/qtl2pleio-on-cran/">Read More…</a></p>
              
            </div>
          </div>
          
          <div class="is-row is-stack-20">
            <div class="is-col is-33 dateblock">
              <h3><a href="/blog/improvboston-101-student-showcase-video/">ImprovBoston Student Showcase video</a></h3>
              <h4>5 January 2020</h4>
            </div>
            <div class="is-col is-66 summary">
              
              


<p>I have now completed <a href="https://improvboston.com/">ImprovBoston’s Improv Comedy 101 class</a>. I’m happy to share the video of my class’s show from December 2019. You can view it <a href="https://venue.streamspot.com/video/9d08388766">here</a>.</p>
<p>(Last modified: 2020-02-08 12:40:41)</p>

              
            </div>
          </div>
          
          <div class="is-row is-stack-20">
            <div class="is-col is-33 dateblock">
              <h3><a href="/blog/internet-social-media/">World-wide Web and Social Media</a></h3>
              <h4>22 December 2018</h4>
            </div>
            <div class="is-col is-66 summary">
              
              I’ve followed a series of blog posts from Cal Newport at http://calnewport.com/blog/ (with posts like this) in which Newport criticizes social media platforms for, among other things, limiting the scope of internet browsing. Newport, a computer science professor and writer, proudly has never had a social media account.
Until March 2018, I regularly consumed information from Facebook, Twitter, and even Instagram and Linkedin. Termination of these accounts aligned with my larger professional and scholarly goals of wanting to focus more on my studies and other, engaging, activities.
              <p class="is-text-right"><a href="/blog/internet-social-media/">Read More…</a></p>
              
            </div>
          </div>
          
          <div class="is-row is-stack-20">
            <div class="is-col is-33 dateblock">
              <h3><a href="/blog/r-package-site-incorporation-into-my-personal-website/">R package site incorporation into my personal website</a></h3>
              <h4>24 November 2018</h4>
            </div>
            <div class="is-col is-66 summary">
              
              Overview I’ve been developing the R package qtl2pleio. It’s now at a stage where others can use it. I recently found the R package pkgdown, which allows developers to build, in an automated fashion, websites for their packages. I posted an entry on pkgdown earlier this year. I now wish to communicate my strategy for incorporating the package site into my personal website, https://fboehm.us. I write this primarily as a reminder to myself.
              <p class="is-text-right"><a href="/blog/r-package-site-incorporation-into-my-personal-website/">Read More…</a></p>
              
            </div>
          </div>
          
          <div class="is-col is-33 dateblock">&nbsp;</div>
          <div class="is-col is-100">
            <p class="is-text-right is-big"><a href="blog">All blog posts&nbsp;<span class="pointer">1</span></a></p>
          </div>
        </div>
        <div class="is-col is-33 is-stack-20">
          <div class="is-row is-stack-20">
            <h2 class="is-center">Recent Works</h2>
          </div>
          <ul class="compact fa-ul">
            
              
              
                
                  <li class="is-inset-bottom-4">

<i class="fa-li fa fa-scroll fa-fw"></i><strong>Genetic Background Modifies Phenotypic Severity and Longevity in a Mouse Model of Niemann-Pick Disease Type C1</strong>.
Jorge L. Rodriguez-Gil, Dawn E. Watkins-Chow, Laura L. Baxter, Gene Elliot, Ursula L. Harper, Stephen M. Wincovitch, Julia C. Wedel, Arturo A. Incao, Mylene Huebecker, Frederick J. Boehm, William S. Garver, Forbes D. Porter, Karl W. Broman, Frances M. Platt and William J. Pavan (2020 )
doi:<a href="https://dx.doi.org/10.1242/dmm.042614">10.1242/dmm.042614</a>



</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-box fa-fw"></i>
   <strong>qtl2pleio:</strong> Testing pleiotropy vs. separate QTL after Jiang &amp; Zeng (1995). (2019)
   &nbsp;<a href="https://github.com/fboehm/qtl2pleio" class="is-smaller"><i class="fab fa-github fa-fw"></i></a>
   &nbsp;<a href="https://cran.r-project.org/web/packages/qtl2pleio/index.html" class="is-smaller"><i class="fab fa-r-project fa-fw"></i></a>
   



  &nbsp;<a data-kube="toggle" data-target="#desc-qtl2pleio" class="is-hidden-print"><i class="fas fa-angle-down fa-fw"></i></a>
  <div id="desc-qtl2pleio" class="is-hidden"><em><code>qtl2pleio</code> performs a d-variate, d-QTL scan over a genomic region of interest before calculating a likelihood ratio test statistic for the competing hypotheses of pleiotropy and separate QTL.</em></div>

</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-chalkboard-teacher fa-fw"></i>
   <strong>Testing pleiotropy in multiparental populations</strong>, <em>Early career scientist seminar, Genetics Society of America, online</em>, Oct 17, 2019
   &nbsp;<a href="https://github.com/fboehm/presentations/tree/master/gsa-early-career-seminar" class="is-smaller"><i class="fas fa-object-ungroup fa-fw"></i></a>
   
   



</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-box fa-fw"></i>
   <strong>gemma2:</strong> Multivariate random effects for systems genetics studies. (2019)
   &nbsp;<a href="https://github.com/fboehm/gemma2" class="is-smaller"><i class="fab fa-github fa-fw"></i></a>
   &nbsp;<a href="https://cran.r-project.org/web/packages/gemma2/index.html" class="is-smaller"><i class="fab fa-r-project fa-fw"></i></a>
   



  &nbsp;<a data-kube="toggle" data-target="#desc-gemma2" class="is-hidden-print"><i class="fas fa-angle-down fa-fw"></i></a>
  <div id="desc-gemma2" class="is-hidden"><em><code>gemma2</code> uses restricted maximum likelihood methods to infer variance components in a multivariate linear mixed effects model for QTL mapping.</em></div>

</li>

                
              
                
                  <li class="is-inset-bottom-4">

<i class="fa-li fa fa-scroll fa-fw"></i><strong>Genetic determinants of gut microbiota composition and bile acid profiles in mice</strong>.
Julia H. Kemis, Vanessa Linke, Kelsey L. Barrett, Frederick J. Boehm, Lindsay L. Traeger, Mark P. Keller, Mary E. Rabaglia, Kathryn L. Schueler, Donald S. Stapleton, Daniel M. Gatti, Gary A. Churchill, Daniel Amador-Noguez, Jason D. Russell, Brian S. Yandell, Karl W. Broman, Joshua J. Coon, Alan D. Attie and Federico E. Rey (2019 )
doi:<a href="https://dx.doi.org/10.1371/journal.pgen.1008073">10.1371/journal.pgen.1008073</a>



</li>

                
              
                
                  <li class="is-inset-bottom-4">

<i class="fa-li fa fa-scroll fa-fw"></i><strong>qtl2pleio: Testing pleiotropy vs. separate QTL in multiparental populations</strong>.
Frederick Boehm, Brian Yandell and Karl Broman (2019 )
doi:<a href="https://dx.doi.org/10.21105/joss.01435">10.21105/joss.01435</a>



</li>

                
              
                
                  <li class="is-inset-bottom-4">

<i class="fa-li fa fa-scroll fa-fw"></i><strong>Testing Pleiotropy vs. Separate QTL in Multiparental Populations</strong>.
Frederick J. Boehm, Elissa J. Chesler, Brian S. Yandell and Karl W. Broman (2019 )
doi:<a href="https://dx.doi.org/10.1534/g3.119.400098">10.1534/g3.119.400098</a>



</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-chalkboard-teacher fa-fw"></i>
   <strong>Testing pleiotropy in multiparental populations</strong>, <em>Thesis defense, University of Wisconsin-Madison, Madison, Wisconsin, USA</em>, Apr 21, 2019
   &nbsp;<a href="https://github.com/fboehm/diss-slides" class="is-smaller"><i class="fas fa-object-ungroup fa-fw"></i></a>
   
   



  &nbsp;<a data-kube="toggle" data-target="#desc-thesis-slides" class="is-hidden-print"><i class="fas fa-angle-down fa-fw"></i></a>
  <div id="desc-thesis-slides" class="is-hidden"><em>Slides for my thesis defense</em></div>

</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-chalkboard-teacher fa-fw"></i>
   <strong>Testing pleiotropy in multiparental populations</strong>, <em>International Mammalian Genomics Conference, Rio Mar, Puerto Rico, USA</em>, Nov 6, 2018
   
   
   



  &nbsp;<a data-kube="toggle" data-target="#desc-imgc-2018" class="is-hidden-print"><i class="fas fa-angle-down fa-fw"></i></a>
  <div id="desc-imgc-2018" class="is-hidden"><em>Awarded best poster honor.</em></div>

</li>

                
              
                
                  <li class="is-inset-bottom-4">

  <i class="fa-li fas fa-chalkboard-teacher fa-fw"></i>
   <strong>Testing pleiotropy in multiparental populations</strong>, <em>Joint Statistical Meetings, Vancouver, British Columbia, Canada</em>, Aug 1, 2018
   &nbsp;<a href="https://github.com/fboehm/jsm2018" class="is-smaller"><i class="fas fa-object-ungroup fa-fw"></i></a>
   
   



  &nbsp;<a data-kube="toggle" data-target="#desc-jsm-2018" class="is-hidden-print"><i class="fas fa-angle-down fa-fw"></i></a>
  <div id="desc-jsm-2018" class="is-hidden"><em>Slide presentation on testing pleiotropy vs. separate QTL in multiparental populations.</em></div>

</li>

                
              
            
            </ul>
          <footer class="group">
            <p class="is-text-right is-big"><a href="vitae">All Works&nbsp;<span class="pointer">1</span></a></p>
          </footer>
        </div>
      </div>
    </div>
 </div>




<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-46757486-5', 'auto');
	
	ga('send', 'pageview');
}
</script>



<footer id="footer" class="is-hidden-print">
  <div class="wrapper">
    <div class="is-row .is-small is-muted">
      <div class="is-col is-33">
        <h3>About</h3>
        <em>
          <p>This is my website. Nothing here is endorsed by my employers, funders, or any organizations I am part of.</p>
          <p>Content on this site is provided under a
            <a href="https://creativecommons.org/licenses/by/4.0/legalcode">Creative Commons (CC-BY) 4.0 license.</a>
            You may re-use this content as long as you indicate Noam Ross's authorship and provide a link back to the original material.
            Source code of the site is provided under the <a href="https://opensource.org/licenses/MIT">MIT license</a>
            and may be reused without restriction.</p>
          <p>This site is gratefully built using a variety of open-source resources and inspirations. Much of the hard work was done by Noam Ross when he built his site. See <a href="https://www.noamross.net/2019/08/09/a-new-website/">here</a> for more about his website and <a href="https://github.com/noamross/noamross.net">here</a> for his source code.</p>
          <p>I've attempted to create a Hugo theme from Noam Ross's materials. The theme is on Github <a href = "https://github.com/fboehm/noamross">here</a>. Please feel free to use and contribute to it.</p>
          <p>I use the Vitruvian Mouse image with permission from <a href="http://daviddeen.com/">David Deen</a>.
</p>
        </em>
      </div>
      <div class="is-col is-33">
        <h3 id="connect">Connect</h3>
        <ul class="fa-ul">
          <li><i class="fa-li fas fa-briefcase fa-fw"></i>I work at <a href="https://www.umassmed.edu/">University of Massachusetts Medical School</a>, Worcester, Massachusetts, USA.</li>
          <li><i class="fa-li fas fa-envelope fa-fw"></i>Email me at <a href="mailto:frederick.boehm@gmail.com">frederick.boehm@gmail.com</a>.</li>
          <li><a href="https://orcid.org/0000-0002-1644-5931"><i class="fa-li ai ai-orcid"></i></a>Track my scholarly work at <a href="https://orcid.org/0000-0002-1644-5931">ORCiD:0000-0002-1644-5931</a>.</li>
          <li><a href="https://scholar.google.com/citations?user=k7PBnv4AAAAJ&hl=en"><i class="fa-li ai ai-google-scholar"></i></a>Or on <a href="https://scholar.google.com/citations?user=k7PBnv4AAAAJ&hl=en">Google Scholar</a>.</li>
          <li><a href="https://github.com/fboehm"><i class="fa-li fab fa-github fa-fw"></i></a>See my open-source code projects at <a href="https://github.com/fboehm">github.com/fboehm</a>.
          <li><a href="/index.xml"><i class="fa-li fas fa-rss fa-fw"></i></a>Follow new posts on this site via the <a href="/index.xml">RSS Feed</a></li>.
        </ul>
      </div>
      <div class="is-col is-33">
        <h3 id="connect">Support</h3>
        <em><p>Some of my work is funded.  In lieu of supporting my volunteer or open-source work, please consider donating to these organizations that I work with or are close to my heart.</p></em>
        <ul class="fa-ul">
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://ropensci.org/donate/">rOpenSci</a> builds tools and community for open science.</li>
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://www.villagehealthworks.org/donate-2">Village Health Works</a> works for health and dignity in Burundi.</li>
          <li><i class="fa-li fas fa-heart fa-fw"></i><a href="https://www.pih.org/pages/donate/give-today">Partners in Health</a> provides health services in many poor communities</li>
        </ul>
      </div>
    </div>
  </div>
</footer>

<script type="text/javascript" src="/assets/js/kube.js"></script>
<script>
  $K.dom(window).on('load', $K.init);
</script>

</body>
</html>
