<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Recla Analysis • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Recla Analysis">
<meta property="og:description" content="qtl2pleio">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-46757486-5"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-46757486-5');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="../articles/d-variate-analysis.html">Multivariate analysis for two or more traits</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="recla-analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Recla Analysis</h1>
                        <h4 class="author">Frederick Boehm</h4>
            
            <h4 class="date">2020-07-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fboehm/qtl2pleio/blob/master/vignettes/recla-analysis.Rmd"><code>vignettes/recla-analysis.Rmd</code></a></small>
      <div class="hidden name"><code>recla-analysis.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">is_inst</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">pkg</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nzchar</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="kw">package</span> <span class="kw">=</span> <span class="no">pkg</span>))
}
<span class="no">qtl2_indic</span> <span class="kw">&lt;-</span> <span class="fu">is_inst</span>(<span class="st">"qtl2"</span>)
<span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span>$<span class="fu">set</span>(<span class="kw">eval</span> <span class="kw">=</span> <span class="no">qtl2_indic</span>)</pre></body></html></div>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">qtl2pleio</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">qtl2</span>)</pre></body></html></div>
<div id="load-recla-data-from-qtl2data-github-repository" class="section level2">
<h2 class="hasAnchor">
<a href="#load-recla-data-from-qtl2data-github-repository" class="anchor"></a>Load Recla data from <code>qtl2data</code> github repository</h2>
<p>We illustrate functions from <code>qtl2pleio</code> by analyzing data from 261 Diversity Outbred mice <span class="citation">(Recla et al. 2014, @logan2013high)</span>.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">file</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,
               <span class="st">"qtl2data/master/DO_Recla/recla.zip"</span>)
<span class="no">recla</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/read_cross2.html">read_cross2</a></span>(<span class="no">file</span>)
<span class="co"># make sex a covariate for use in qtl2pleio::scan_pvl</span>
<span class="no">recla</span><span class="kw">[[</span><span class="fl">6</span>]][ , <span class="fl">1</span>, <span class="kw">drop</span> <span class="kw">=</span> <span class="fl">FALSE</span>] <span class="kw">-&gt;</span> <span class="no">sex</span>
<span class="co"># insert pseudomarkers</span>
<span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/insert_pseudomarkers.html">insert_pseudomarkers</a></span>(<span class="no">recla</span>, <span class="kw">step</span> <span class="kw">=</span> <span class="fl">0.10</span>) <span class="kw">-&gt;</span> <span class="no">pseudomap</span>
<span class="no">gm</span> <span class="kw">&lt;-</span> <span class="no">pseudomap</span>$<span class="no">`8`</span></pre></body></html></div>
<p>We use the hidden Markov model from <span class="citation">Broman (2012a)</span> and <span class="citation">Broman (2012b)</span> (as implemented in <span class="citation">Broman et al. (2019)</span>) to calculate 36-state genotype probabilities for autosomal markers.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">probs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/calc_genoprob.html">calc_genoprob</a></span>(<span class="no">recla</span>, <span class="kw">map</span> <span class="kw">=</span> <span class="no">pseudomap</span>, <span class="kw">cores</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<p>We then convert the genotype probabilities to haplotype dosages.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">aprobs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/genoprob_to_alleleprob.html">genoprob_to_alleleprob</a></span>(<span class="no">probs</span>)</pre></body></html></div>
<p>We calculate kinship matrices, by the “leave one chromosome out (loco)” method.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">kinship</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/calc_kinship.html">calc_kinship</a></span>(<span class="no">aprobs</span>, <span class="st">"loco"</span>)</pre></body></html></div>
<p>Before performing our QTL mapping, we transform the phenotypes.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">recla</span>$<span class="no">pheno</span> <span class="kw">-&gt;</span> <span class="no">ph</span>
<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span>(<span class="no">ph</span>) <span class="kw">-&gt;</span> <span class="no">lph</span>
<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="kw">FUN</span> <span class="kw">=</span> <span class="kw pkg">broman</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/broman/man/winsorize.html">winsorize</a></span>, <span class="kw">X</span> <span class="kw">=</span> <span class="no">lph</span>, <span class="kw">MARGIN</span> <span class="kw">=</span> <span class="fl">2</span>) <span class="kw">-&gt;</span> <span class="no">wlph</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>(<span class="no">wlph</span>) <span class="kw">-&gt;</span> <span class="no">wlph_tib</span></pre></body></html></div>
<p>We next perform the univariate QTL scan for all phenotypes.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">sex2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="no">sex</span> <span class="kw">==</span> <span class="st">"female"</span>), <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">sex2</span>) <span class="kw">&lt;-</span> <span class="st">"female"</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sex2</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">aprobs</span><span class="kw">[[</span><span class="fl">1</span>]])
<span class="no">out</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/scan1.html">scan1</a></span>(<span class="kw">genoprobs</span> <span class="kw">=</span> <span class="no">aprobs</span>,
             <span class="kw">pheno</span> <span class="kw">=</span> <span class="no">wlph</span>,
             <span class="kw">kinship</span> <span class="kw">=</span> <span class="no">kinship</span>,
             <span class="kw">addcovar</span> <span class="kw">=</span> <span class="no">sex2</span>,
             <span class="kw">reml</span> <span class="kw">=</span> <span class="fl">TRUE</span>
             )</pre></body></html></div>
<p>We want to look closely at those peaks on Chromosome 8. We’ll save the positions of peaks for our two traits of interest.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r">(<span class="no">peaks</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/find_peaks.html">find_peaks</a></span>(<span class="no">out</span>, <span class="no">pseudomap</span>, <span class="kw">threshold</span> <span class="kw">=</span> <span class="fl">5</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="no">chr</span>, <span class="no">pos</span>) <span class="kw">%&gt;%</span>
   <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(- <span class="no">lodindex</span>))</pre></body></html></div>
<pre><code>##                    lodcolumn chr     pos      lod
## 1                         bw   1 23.9075 5.471580
## 2         OF_distance_first4   1 43.2385 5.772977
## 3             LD_transitions   1 95.8075 5.028258
## 4                OF_distance   2 49.9770 5.544458
## 5                         bw   2 52.3932 7.352334
## 6            OF_immobile_pct   2 53.2646 9.771814
## 7  VC_bottom_distance_first4   2 71.0160 6.531654
## 8         OF_distance_first4   3 10.7360 5.541166
## 9  VC_bottom_distance_first4   3 16.3700 5.518637
## 10           VC_top_time_pct   3 17.9390 5.951067
## 11         LD_distance_light   3 23.4390 5.203246
## 12                        bw   3 24.8390 5.632714
## 13        VC_top_time_first4   3 48.1280 6.144073
## 14           VC_top_velocity   3 48.5630 6.264824
## 15        VC_top_time_first4   4  3.5340 5.016970
## 16             OF_corner_pct   4  9.0111 6.272996
## 17           OF_immobile_pct   4 37.5206 5.529985
## 18         LD_distance_light   4 71.2992 5.040185
## 19                        bw   5 10.0740 5.728865
## 20 VC_bottom_distance_first4   5 19.9741 5.419061
## 21        VC_top_time_first4   5 20.0741 6.075282
## 22        VC_bottom_distance   5 20.5930 5.691913
## 23        VC_bottom_time_pct   5 20.5930 6.360001
## 24       TS_latency_immobile   5 43.3504 5.995115
## 25             OF_corner_pct   5 64.2551 5.665658
## 26           OF_immobile_pct   6 53.4292 6.968771
## 27     TS_frequency_climbing   6 57.0362 5.361091
## 28                        bw   7  9.1778 6.057098
## 29          TS_time_immobile   7 49.4778 8.067612
## 30        VC_bottom_distance   7 54.5591 5.011113
## 31        OF_distance_first4   7 57.9454 5.327302
## 32           VC_top_distance   7 83.8778 5.724788
## 33           OF_immobile_pct   7 83.9778 5.823555
## 34     TS_frequency_climbing   8 48.1732 5.483064
## 35         LD_distance_light   8 55.2762 5.323391
## 36              LD_light_pct   8 55.2762 5.274185
## 37                HP_latency   8 57.7732 6.223739
## 38         LD_distance_light   9 36.6965 5.196968
## 39              LD_light_pct   9 36.6965 5.417419
## 40        VC_top_time_first4   9 38.4834 5.109813
## 41           VC_top_time_pct   9 39.2680 6.356432
## 42                HP_latency   9 46.8502 5.222074
## 43                        bw  10  3.7781 6.526199
## 44        OF_distance_first4  10 29.6698 5.462975
## 45        VC_bottom_time_pct  10 32.5438 5.432804
## 46          OF_periphery_pct  10 74.8530 5.246487
## 47           VC_top_distance  11  7.8200 6.245803
## 48    VC_top_distance_first4  11 11.6236 5.486915
## 49 VC_bottom_distance_first4  11 54.3420 5.367052
## 50            LD_transitions  11 58.9000 5.903217
## 51     VC_bottom_transitions  11 60.5984 5.114051
## 52              LD_light_pct  11 63.3943 6.464176
## 53         LD_distance_light  11 63.4514 6.373437
## 54           VC_top_time_pct  12 20.5776 6.950143
## 55        VC_bottom_velocity  12 21.7760 5.653292
## 56             OF_center_pct  12 35.5140 6.399422
## 57                HP_latency  12 43.5150 5.131074
## 58          OF_periphery_pct  12 53.5776 7.240951
## 59             OF_corner_pct  13 59.7966 6.594594
## 60     VC_bottom_time_first4  14 11.9183 5.204368
## 61 VC_bottom_distance_first4  14 12.5316 5.763233
## 62     VC_bottom_transitions  14 12.5316 6.478533
## 63           VC_top_velocity  14 12.7819 6.840412
## 64        VC_bottom_distance  14 14.5316 5.592029
## 65     TS_frequency_climbing  14 21.1141 5.372594
## 66             OF_center_pct  14 53.7316 5.377182
## 67     TS_frequency_climbing  15 12.6680 6.043265
## 68              LD_light_pct  15 15.2374 5.674921
## 69        OF_distance_first4  16 23.2656 5.242102
## 70           VC_top_distance  17 15.6390 6.669277
## 71           VC_top_velocity  18  8.3750 5.558628
## 72        VC_top_time_first4  18 17.8068 6.246206
## 73            LD_transitions  18 37.4182 5.090332
## 74 VC_bottom_distance_first4  19 24.9615 7.362557
## 75     VC_bottom_time_first4  19 24.9615 7.499959
## 76           OF_immobile_pct  19 31.9505 5.639812
## 77                HP_latency  19 47.7977 5.485000</code></pre>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">peaks8</span> <span class="kw">&lt;-</span> <span class="no">peaks</span> <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">chr</span> <span class="kw">==</span> <span class="fl">8</span>, <span class="no">pos</span> <span class="kw">&gt;</span> <span class="fl">50</span>, <span class="no">pos</span> <span class="kw">&lt;</span> <span class="fl">60</span>)
<span class="no">pos_LD_light_pct</span> <span class="kw">&lt;-</span> <span class="no">peaks8</span> <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">lodcolumn</span> <span class="kw">==</span> <span class="st">"LD_light_pct"</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">pos</span>)
<span class="no">pos_HP_latency</span> <span class="kw">&lt;-</span> <span class="no">peaks8</span> <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">lodcolumn</span> <span class="kw">==</span> <span class="st">"HP_latency"</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">pos</span>)</pre></body></html></div>
</div>
<div id="correlation" class="section level2">
<h2 class="hasAnchor">
<a href="#correlation" class="anchor"></a>Correlation</h2>
<p>Given that the two traits “percent time in light” and “distance traveled in light” share a peak, we want to ask how correlated they are.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span>(<span class="no">wlph</span>[ , <span class="fl">7</span>], <span class="no">wlph</span>[ , <span class="fl">10</span>], <span class="kw">use</span> <span class="kw">=</span> <span class="st">"complete.obs"</span>)</pre></body></html></div>
<pre><code>## [1] 0.8859402</code></pre>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span>(<span class="no">wlph</span>[ , <span class="fl">22</span>], <span class="no">wlph</span>[ , <span class="fl">10</span>], <span class="kw">use</span> <span class="kw">=</span> <span class="st">"complete.obs"</span>)</pre></body></html></div>
<pre><code>## [1] -0.1507317</code></pre>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span>(<span class="no">wlph</span>[ , <span class="fl">7</span>], <span class="no">wlph</span>[ , <span class="fl">22</span>], <span class="kw">use</span> <span class="kw">=</span> <span class="st">"complete.obs"</span>)</pre></body></html></div>
<pre><code>## [1] -0.143598</code></pre>
<p>Since “percent time in light” and “distance traveled in light” are very highly correlated, we’ll discard “distance traveled in light” and perform subsequent analyses with only “percent time in light” and the second trait, “hot plate latency”.</p>
</div>
<div id="scatter-plot-of-phenotypes" class="section level2">
<h2 class="hasAnchor">
<a href="#scatter-plot-of-phenotypes" class="anchor"></a>Scatter plot of phenotypes</h2>
<p>We create a scatter plot for the two phenotypes, “hot plate latency” and “percent time in light”.</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">wlph_tib</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">y</span> <span class="kw">=</span> <span class="no">HP_latency</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="no">LD_light_pct</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Percent time in light"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Hot plate latency"</span>)</pre></body></html></div>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="recla-analysis_files/figure-html/scatter-1.png" width="700"></p>
</div>
<div id="genome-wide-lod-plots-for-the-traits-from-recla" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-wide-lod-plots-for-the-traits-from-recla" class="anchor"></a>Genome-wide LOD plots for the traits from Recla</h2>
<p>Let’s plot the results of the univariate QTL scans for our two traits.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">out</span>, <span class="kw">map</span> <span class="kw">=</span> <span class="no">pseudomap</span>,
     <span class="kw">lodcolumn</span> <span class="kw">=</span> <span class="fl">10</span>,
     <span class="kw">main</span> <span class="kw">=</span> <span class="st">"percent time in light"</span>
     )</pre></body></html></div>
<p><img src="recla-analysis_files/figure-html/lod10-plot-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">out</span>, <span class="kw">map</span> <span class="kw">=</span> <span class="no">pseudomap</span>,
     <span class="kw">lodcolumn</span> <span class="kw">=</span> <span class="fl">22</span>,
     <span class="kw">main</span> <span class="kw">=</span> <span class="st">"hot plate latency"</span>
     )</pre></body></html></div>
<p><img src="recla-analysis_files/figure-html/lod22-plot-1.png" width="700"></p>
</div>
<div id="allele-effects-plots-on-chr-8-for-each-of-the-three-recla-traits" class="section level2">
<h2 class="hasAnchor">
<a href="#allele-effects-plots-on-chr-8-for-each-of-the-three-recla-traits" class="anchor"></a>Allele effects plots on Chr 8 for each of the three Recla traits</h2>
<p>We examine the allele effects plots for our two traits, in the region of interest on Chromosome 8.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/scan1coef.html">scan1coef</a></span>(<span class="no">aprobs</span>[ , <span class="fl">8</span>], <span class="kw">pheno</span> <span class="kw">=</span> <span class="no">wlph</span>[ , <span class="fl">10</span>], <span class="kw">kinship</span> <span class="kw">=</span> <span class="no">kinship</span>$<span class="no">`8`</span>,
          <span class="kw">reml</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
          <span class="kw">addcovar</span> <span class="kw">=</span> <span class="no">sex2</span>) <span class="kw">-&gt;</span> <span class="no">s1c_10</span>
<span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/scan1coef.html">scan1coef</a></span>(<span class="no">aprobs</span>[ , <span class="fl">8</span>], <span class="kw">pheno</span> <span class="kw">=</span> <span class="no">wlph</span>[ , <span class="fl">22</span>], <span class="kw">kinship</span> <span class="kw">=</span> <span class="no">kinship</span>$<span class="no">`8`</span>,
          <span class="kw">reml</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
          <span class="kw">addcovar</span> <span class="kw">=</span> <span class="no">sex2</span>) <span class="kw">-&gt;</span> <span class="no">s1c_22</span></pre></body></html></div>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="co"># subset scan1output objects</span>
<span class="no">s1c_10s</span> <span class="kw">&lt;-</span> <span class="no">s1c_10</span>[<span class="fl">650</span>:<span class="fl">999</span>, ]
<span class="co"># 650:999 is the same as the interval for the two-dimensional scan.</span>
<span class="no">s1c_22s</span> <span class="kw">&lt;-</span> <span class="no">s1c_22</span>[<span class="fl">650</span>:<span class="fl">999</span>, ]</pre></body></html></div>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/plot_coef.html">plot_coefCC</a></span>(<span class="no">s1c_10s</span>, <span class="kw">scan1_output</span> <span class="kw">=</span> <span class="no">out</span>[ , <span class="fl">10</span>, <span class="kw">drop</span> <span class="kw">=</span> <span class="fl">FALSE</span>], <span class="kw">map</span> <span class="kw">=</span> <span class="no">pseudomap</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"percent time in light"</span>)</pre></body></html></div>
<p><img src="recla-analysis_files/figure-html/plot-coefs-1.png" width="700"></p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/qtl2/man/plot_coef.html">plot_coefCC</a></span>(<span class="no">s1c_22s</span>, <span class="kw">scan1_output</span> <span class="kw">=</span> <span class="no">out</span>[ , <span class="fl">22</span>, <span class="kw">drop</span> <span class="kw">=</span> <span class="fl">FALSE</span>], <span class="kw">map</span> <span class="kw">=</span> <span class="no">pseudomap</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"hot plate latency"</span>)</pre></body></html></div>
<p><img src="recla-analysis_files/figure-html/plot-coefs-2.png" width="700"></p>
</div>
<div id="two-dimensional-scan-results-from-github" class="section level2">
<h2 class="hasAnchor">
<a href="#two-dimensional-scan-results-from-github" class="anchor"></a>Two-dimensional scan results from github</h2>
<p>We present the code that we ran to perform the two-dimensional scan.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="../reference/scan_pvl.html">scan_pvl</a></span>(<span class="kw">probs</span> <span class="kw">=</span> <span class="no">aprobs</span>$<span class="no">`8`</span>,
         <span class="kw">pheno</span> <span class="kw">=</span> <span class="no">wlph</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">10</span>, <span class="fl">22</span>)],
         <span class="kw">addcovar</span> <span class="kw">=</span> <span class="no">sex2</span>,
         <span class="kw">kinship</span> <span class="kw">=</span> <span class="no">kinship</span>$<span class="no">`8`</span>,
         <span class="kw">start_snp</span> <span class="kw">=</span> <span class="fl">650</span>,
         <span class="kw">n_snp</span> <span class="kw">=</span> <span class="fl">350</span>) <span class="kw">-&gt;</span> <span class="no">pvl1022</span></pre></body></html></div>
<p>To save computing time, we read the two-dimensional scan results from Zenodo. Because the Zenodo file was created with an older version of <code>scan_pvl</code>, it contains the log (natural base) likelihoods, rather than log10 likelihoods. Thus, we’ll want to rectify this, which we do with code below.</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span>(<span class="st">"https://zenodo.org/record/3210710/files/recla-10-22.txt?download=1"</span>, <span class="kw">stringsAsFactors</span> <span class="kw">=</span> <span class="fl">FALSE</span>)) <span class="kw">-&gt;</span> <span class="no">pvl1022</span></pre></body></html></div>
<p>We then calculate the likelihood ratio test statistic.</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r">(<span class="no">mylrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/calc_lrt_tib.html">calc_lrt_tib</a></span>(<span class="no">pvl1022</span>))</pre></body></html></div>
<pre><code>## [1] 2.771408</code></pre>
</div>
<div id="profile-lod-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#profile-lod-plot" class="anchor"></a>Profile LOD Plot</h2>
<p>We create a profile LOD plot.</p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">recla</span>$<span class="no">pheno</span>)[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">10</span>, <span class="fl">22</span>)] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Percent time in light"</span>, <span class="st">"Hot plate latency"</span>)
<span class="no">pvl1022</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">log10lik</span> <span class="kw">=</span> <span class="no">loglik</span> / <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span>(<span class="fl">10</span>)) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(- <span class="no">loglik</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="../reference/calc_profile_lods.html">calc_profile_lods</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="../reference/add_pmap.html">add_pmap</a></span>(<span class="kw">pmap</span> <span class="kw">=</span> <span class="no">recla</span>$<span class="no">pmap</span>$<span class="no">`8`</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>() + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">marker_position</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">profile_lod</span>, <span class="kw">colour</span> <span class="kw">=</span> <span class="no">trait</span>))</pre></body></html></div>
<pre><code>## Joining, by = "marker"</code></pre>
<p><img src="recla-analysis_files/figure-html/profile-plot-1.png" width="700"></p>
</div>
<div id="bootstrap-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#bootstrap-analyses" class="anchor"></a>Bootstrap analyses</h2>
<p>First, we find the <em>pleiotropy peak</em> marker. This is the marker for which the log likelihood is maximized under the constraint of pleiotropy.</p>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu"><a href="../reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span>(<span class="no">pvl1022</span>, <span class="kw">start_snp</span> <span class="kw">=</span> <span class="fl">650</span>)</pre></body></html></div>
<pre><code>## loglik139 
##       788</code></pre>
<p>We need the <em>pleiotropy peak</em> marker in the bootstrap analyses because it is the marker used in drawing bootstrap samples.</p>
<p>To save computing time, we read the bootstrap results files from Github. For details of how we performed the bootstrap analyses on the University of Wisconsin-Madison Center for High-Throughput Computing, please see the documentation in the qtl2pleio-manuscript repository: <a href="https://github.com/fboehm/qtl2pleio-manuscript" class="uri">https://github.com/fboehm/qtl2pleio-manuscript</a>.</p>
<p>The code below creates a temporary directory “tmp” in the user’s working directory. We then read results files from Github. Each text file contains a single likelihood ratio test statistic from a bootstrap sample.</p>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="co">## read boot lrt files</span>
<span class="no">boot_lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:<span class="fl">1000</span>){
  <span class="no">n</span> <span class="kw">&lt;-</span> <span class="no">i</span> - <span class="fl">1</span>
  <span class="no">fn</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/fboehm/qtl2pleio-manuscript-chtc/master/Recla-bootstrap/results/recla-boot-run561_"</span>, <span class="no">n</span>, <span class="st">".txt"</span>)
  <span class="no">boot_lrt</span>[<span class="no">i</span>] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span>(<span class="no">fn</span>)
}
<span class="co"># convert list to numeric vector</span>
<span class="no">boot_lrt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">boot_lrt</span>)</pre></body></html></div>
<p>We get a bootstrap p-value by comparing the above vector’s values to <code>mylrt</code>, the test statistic for the observed data.</p>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">boot_lrt</span> <span class="kw">&gt;=</span> <span class="no">mylrt</span>) / <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">boot_lrt</span>)</pre></body></html></div>
<pre><code>## [1] 0.109</code></pre>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/reexports.html">session_info</a></span>()</pre></body></html></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.0 (2020-04-24)
##  os       Ubuntu 20.04 LTS            
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       America/New_York            
##  date     2020-07-01                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package     * version  date       lib source        
##  assertthat    0.2.1    2019-03-21 [2] CRAN (R 4.0.0)
##  backports     1.1.8    2020-06-17 [2] CRAN (R 4.0.0)
##  bit           1.1-15.2 2020-02-10 [2] CRAN (R 4.0.0)
##  bit64         0.9-7    2017-05-08 [2] CRAN (R 4.0.0)
##  blob          1.2.1    2020-01-20 [2] CRAN (R 4.0.0)
##  broman        0.69-5   2019-04-11 [2] CRAN (R 4.0.0)
##  callr         3.4.3    2020-03-28 [2] CRAN (R 4.0.0)
##  cli           2.0.2    2020-02-28 [2] CRAN (R 4.0.0)
##  colorspace    1.4-1    2019-03-18 [2] CRAN (R 4.0.0)
##  crayon        1.3.4    2017-09-16 [2] CRAN (R 4.0.0)
##  data.table    1.12.8   2019-12-09 [2] CRAN (R 4.0.0)
##  DBI           1.1.0    2019-12-15 [2] CRAN (R 4.0.0)
##  desc          1.2.0    2018-05-01 [2] CRAN (R 4.0.0)
##  devtools      2.3.0    2020-04-10 [2] CRAN (R 4.0.0)
##  digest        0.6.25   2020-02-23 [2] CRAN (R 4.0.0)
##  dplyr       * 1.0.0    2020-05-29 [2] CRAN (R 4.0.0)
##  ellipsis      0.3.1    2020-05-15 [2] CRAN (R 4.0.0)
##  evaluate      0.14     2019-05-28 [2] CRAN (R 4.0.0)
##  fansi         0.4.1    2020-01-08 [2] CRAN (R 4.0.0)
##  farver        2.0.3    2020-01-16 [2] CRAN (R 4.0.0)
##  fs            1.4.1    2020-04-04 [2] CRAN (R 4.0.0)
##  generics      0.0.2    2018-11-29 [2] CRAN (R 4.0.0)
##  ggplot2     * 3.3.2    2020-06-19 [2] CRAN (R 4.0.0)
##  glue          1.4.1    2020-05-13 [2] CRAN (R 4.0.0)
##  gtable        0.3.0    2019-03-25 [2] CRAN (R 4.0.0)
##  htmltools     0.5.0    2020-06-16 [2] CRAN (R 4.0.0)
##  jsonlite      1.6.1    2020-02-02 [2] CRAN (R 4.0.0)
##  knitr         1.28     2020-02-06 [2] CRAN (R 4.0.0)
##  labeling      0.3      2014-08-23 [2] CRAN (R 4.0.0)
##  lifecycle     0.2.0    2020-03-06 [2] CRAN (R 4.0.0)
##  magrittr      1.5      2014-11-22 [2] CRAN (R 4.0.0)
##  MASS          7.3-51.6 2020-04-26 [2] CRAN (R 4.0.0)
##  memoise       1.1.0    2017-04-21 [2] CRAN (R 4.0.0)
##  munsell       0.5.0    2018-06-12 [2] CRAN (R 4.0.0)
##  pillar        1.4.4    2020-05-05 [2] CRAN (R 4.0.0)
##  pkgbuild      1.0.8    2020-05-07 [2] CRAN (R 4.0.0)
##  pkgconfig     2.0.3    2019-09-22 [2] CRAN (R 4.0.0)
##  pkgdown       1.5.1    2020-04-09 [2] CRAN (R 4.0.0)
##  pkgload       1.1.0    2020-05-29 [2] CRAN (R 4.0.0)
##  prettyunits   1.1.1    2020-01-24 [2] CRAN (R 4.0.0)
##  processx      3.4.2    2020-02-09 [2] CRAN (R 4.0.0)
##  ps            1.3.3    2020-05-08 [2] CRAN (R 4.0.0)
##  purrr         0.3.4    2020-04-17 [2] CRAN (R 4.0.0)
##  qtl2        * 0.22-8   2020-06-26 [2] CRAN (R 4.0.0)
##  qtl2pleio   * 1.3.0    2020-07-01 [1] local         
##  R6            2.4.1    2019-11-12 [2] CRAN (R 4.0.0)
##  Rcpp          1.0.4.6  2020-04-09 [2] CRAN (R 4.0.0)
##  remotes       2.1.1    2020-02-15 [2] CRAN (R 4.0.0)
##  rlang         0.4.6    2020-05-02 [2] CRAN (R 4.0.0)
##  rmarkdown     2.3      2020-06-18 [2] CRAN (R 4.0.0)
##  rprojroot     1.3-2    2018-01-03 [2] CRAN (R 4.0.0)
##  RSQLite       2.2.0    2020-01-07 [2] CRAN (R 4.0.0)
##  rstudioapi    0.11     2020-02-07 [2] CRAN (R 4.0.0)
##  scales        1.1.1    2020-05-11 [2] CRAN (R 4.0.0)
##  sessioninfo   1.1.1    2018-11-05 [2] CRAN (R 4.0.0)
##  stringi       1.4.6    2020-02-17 [2] CRAN (R 4.0.0)
##  stringr       1.4.0    2019-02-10 [2] CRAN (R 4.0.0)
##  testthat      2.3.2    2020-03-02 [2] CRAN (R 4.0.0)
##  tibble        3.0.1    2020-04-20 [2] CRAN (R 4.0.0)
##  tidyselect    1.1.0    2020-05-11 [2] CRAN (R 4.0.0)
##  usethis       1.6.1    2020-04-29 [2] CRAN (R 4.0.0)
##  vctrs         0.3.1    2020-06-05 [2] CRAN (R 4.0.0)
##  withr         2.2.0    2020-04-20 [2] CRAN (R 4.0.0)
##  xfun          0.15     2020-06-21 [2] CRAN (R 4.0.0)
##  yaml          2.2.1    2020-02-01 [2] CRAN (R 4.0.0)
## 
## [1] /tmp/RtmpOzXNnc/temp_libpath130b0529daeb1
## [2] /home/fred/R/x86_64-pc-linux-gnu-library/4.0
## [3] /usr/local/lib/R/site-library
## [4] /usr/lib/R/site-library
## [5] /usr/lib/R/library</code></pre>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-broman2012genotype">
<p>Broman, Karl W. 2012a. “Genotype Probabilities at Intermediate Generations in the Construction of Recombinant Inbred Lines.” <em>Genetics</em> 190 (2): 403–12.</p>
</div>
<div id="ref-broman2012haplotype">
<p>———. 2012b. “Haplotype Probabilities in Advanced Intercross Populations.” <em>G3: Genes, Genomes, Genetics</em> 2 (2): 199–202.</p>
</div>
<div id="ref-broman2019rqtl2">
<p>Broman, Karl W, Daniel M Gatti, Petr Simecek, Nicholas A Furlotte, Pjotr Prins, Śaunak Sen, Brian S Yandell, and Gary A Churchill. 2019. “R/Qtl2: Software for Mapping Quantitative Trait Loci with High-Dimensional Data and Multiparent Populations.” <em>Genetics</em> 211 (2): 495–502.</p>
</div>
<div id="ref-logan2013high">
<p>Logan, Ryan W, Raymond F Robledo, Jill M Recla, Vivek M Philip, Jason A Bubier, Jeremy J Jay, Carter Harwood, et al. 2013. “High-Precision Genetic Mapping of Behavioral Traits in the Diversity Outbred Mouse Population.” <em>Genes, Brain and Behavior</em> 12 (4): 424–37.</p>
</div>
<div id="ref-recla2014precise">
<p>Recla, Jill M, Raymond F Robledo, Daniel M Gatti, Carol J Bult, Gary A Churchill, and Elissa J Chesler. 2014. “Precise Genetic Mapping and Integrative Bioinformatics in Diversity Outbred Mice Reveals Hydin as a Novel Pain Gene.” <em>Mammalian Genome</em> 25 (5-6): 211–22.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Frederick J Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
