<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Multivariate analysis for two or more traits • qtl2pleio</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Multivariate analysis for two or more traits" />
<meta property="og:description" content="qtl2pleio" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-46757486-5"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-46757486-5');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="../articles/d-variate-analysis.html">Multivariate analysis for two or more traits</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fboehm/qtl2pleio/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<script src="d-variate-analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Multivariate analysis for two or more traits</h1>
                        <h4 class="author">Frederick Boehm</h4>
            
            <h4 class="date">2020-07-08</h4>
      
      <small class="dont-index">Source: <a href='https://github.com/fboehm/qtl2pleio/blob/master/vignettes/d-variate-analysis.Rmd'><code>vignettes/d-variate-analysis.Rmd</code></a></small>
      <div class="hidden name"><code>d-variate-analysis.Rmd</code></div>

    </div>

    
    
<p>The goal of this vignette is to introduce the functions that enable d-variate (for <span class="math inline">\(d \ge 2\)</span>) QTL analysis for d traits that each univariately map to a shared region of the genome. One may then ask whether the d traits share a single, pleiotropic QTL. The alternative hypothesis is that there is more than one QTL giving rise to the phenotype-genotype associations.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>is_inst &lt;-<span class="st"> </span><span class="cf">function</span>(pkg) {</span>
<span id="cb1-2"><a href="#cb1-2"></a>    <span class="kw">nzchar</span>(<span class="kw">system.file</span>(<span class="dt">package =</span> pkg))</span>
<span id="cb1-3"><a href="#cb1-3"></a>}</span>
<span id="cb1-4"><a href="#cb1-4"></a>qtl2_indic &lt;-<span class="st"> </span><span class="kw">is_inst</span>(<span class="st">&quot;qtl2&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(qtl2pleio)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">library</span>(qtl2)</span></code></pre></div>
<p>We also load the <code>qtl2</code> package with the <code>library</code> command.</p>
<div id="reading-data-from-qtl2data-repository-on-github" class="section level2">
<h2>Reading data from <code>qtl2data</code> repository on github</h2>
<p>We’ll consider the <a href="https://github.com/rqtl/qtl2data/tree/master/DOex"><code>DOex</code></a> data in the <a href="https://github.com/rqtl/qtl2data"><code>qtl2data</code></a> repository. We’ll download the DOex.zip file before calculating founder allele dosages.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>file &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;https://raw.githubusercontent.com/rqtl/&quot;</span>,</span>
<span id="cb3-2"><a href="#cb3-2"></a>               <span class="st">&quot;qtl2data/master/DOex/DOex.zip&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a>DOex &lt;-<span class="st"> </span><span class="kw">read_cross2</span>(file)</span></code></pre></div>
<p>Let’s calculate the founder allele dosages from the 36-state genotype probabilities.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>probs &lt;-<span class="st"> </span><span class="kw">calc_genoprob</span>(DOex)</span>
<span id="cb4-2"><a href="#cb4-2"></a>pr &lt;-<span class="st"> </span><span class="kw">genoprob_to_alleleprob</span>(probs)</span></code></pre></div>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
</div>
<div id="kinship-calculations" class="section level2">
<h2>Kinship calculations</h2>
<p>For our statistical model, we need a kinship matrix. Although we don’t have genome-wide data - since we have allele probabilities for only 3 chromosomes - let’s calculate a kinship matrix using “leave-one-chromosome-out”. In practice, one would want to use allele probabilities from a full genome-wide set of markers.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">calc_kinship</span>(<span class="dt">probs =</span> pr, <span class="dt">type =</span> <span class="st">&quot;loco&quot;</span>) -&gt;<span class="st"> </span>kinship</span></code></pre></div>
</div>
<div id="simulating-phenotypes-with-qtl2pleiosim1" class="section level2">
<h2>Simulating phenotypes with <code>qtl2pleio::sim1</code></h2>
<p>The function to simulate phenotypes in <code>qtl2pleio</code> is <code>sim1</code>. By examining its help page, we see that it takes five arguments. The help page also gives the dimensions of the inputs.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># set up the design matrix, X</span></span>
<span id="cb6-2"><a href="#cb6-2"></a>pp &lt;-<span class="st"> </span>pr[[<span class="dv">2</span>]] <span class="co">#we&#39;ll work with Chr 3&#39;s genotype data</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="kw">dim</span>(pp)</span></code></pre></div>
<pre><code>## [1] 261   8 102</code></pre>
<p>We prepare a block-diagonal design matrix X that contains two nonzero blocks on the diagonal, one for each trait. We use here a function from the <code>gemma2</code> R package to set up the needed matrix.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">#Next, we prepare a design matrix X</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>X &lt;-<span class="st"> </span>gemma2<span class="op">::</span><span class="kw">stagger_mats</span>(pp[ , , <span class="dv">50</span>], pp[ , , <span class="dv">50</span>], pp[ , , <span class="dv">50</span>])</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw">dim</span>(X)</span></code></pre></div>
<pre><code>## [1] 783  24</code></pre>
<p><span class="math inline">\(X\)</span> is a block-diagonal matrix, with 3 diagonal blocks of equal dimension.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># assemble B matrix of allele effects</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>B &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">data =</span> <span class="kw">rep</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="dt">each =</span> <span class="dv">4</span>), <span class="dt">times =</span> <span class="dv">3</span>), <span class="dt">nrow =</span> <span class="dv">8</span>, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="co"># verify that B is what we want:</span></span>
<span id="cb10-4"><a href="#cb10-4"></a>B</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]   -1   -1   -1
## [2,]   -1   -1   -1
## [3,]   -1   -1   -1
## [4,]   -1   -1   -1
## [5,]    1    1    1
## [6,]    1    1    1
## [7,]    1    1    1
## [8,]    1    1    1</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># set.seed to ensure reproducibility</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">set.seed</span>(<span class="dv">2018-01-30</span>)</span>
<span id="cb12-3"><a href="#cb12-3"></a>Sigma &lt;-<span class="st"> </span><span class="kw">calc_Sigma</span>(<span class="dt">Vg =</span> <span class="kw">diag</span>(<span class="dv">3</span>), <span class="dt">Ve =</span> <span class="kw">diag</span>(<span class="dv">3</span>), <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]])</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co"># call to sim1</span></span>
<span id="cb12-5"><a href="#cb12-5"></a>Ypre &lt;-<span class="st"> </span><span class="kw">sim1</span>(<span class="dt">X =</span> X, <span class="dt">B =</span> B, <span class="dt">Sigma =</span> Sigma)</span>
<span id="cb12-6"><a href="#cb12-6"></a>Y &lt;-<span class="st"> </span><span class="kw">matrix</span>(Ypre, <span class="dt">nrow =</span> <span class="dv">261</span>, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="kw">rownames</span>(Y) &lt;-<span class="st"> </span><span class="kw">rownames</span>(pp)</span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="kw">colnames</span>(Y) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;tr1&quot;</span>, <span class="st">&quot;tr2&quot;</span>, <span class="st">&quot;tr3&quot;</span>)</span></code></pre></div>
<p>Let’s perform univariate QTL mapping for each trait (ie, each column) in the Y matrix.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>s1 &lt;-<span class="st"> </span><span class="kw">scan1</span>(<span class="dt">genoprobs =</span> pr, <span class="dt">pheno =</span> Y, <span class="dt">kinship =</span> kinship)</span></code></pre></div>
<p>Here is a plot of the results.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">plot</span>(s1, DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>)</span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="kw">plot</span>(s1, DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, <span class="dt">lod =</span> <span class="dv">2</span>, <span class="dt">col =</span><span class="st">&quot;violetred&quot;</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="kw">plot</span>(s1, DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, <span class="dt">lod =</span> <span class="dv">3</span>, <span class="dt">col =</span><span class="st">&quot;green&quot;</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="kw">colnames</span>(s1), <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;darkslateblue&quot;</span>, <span class="st">&quot;violetred&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="dt">bg=</span><span class="st">&quot;gray92&quot;</span>)</span></code></pre></div>
<p><img src="d-variate-analysis_files/figure-html/1d-lod-plots-1.png" width="700" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw">find_peaks</span>(s1, <span class="dt">map =</span> DOex<span class="op">$</span>pmap, <span class="dt">threshold =</span> <span class="dv">8</span>)</span></code></pre></div>
<pre><code>##   lodindex lodcolumn chr      pos      lod
## 1        1       tr1   3 81.75897 17.60763
## 2        2       tr2   3 82.77806 19.82599
## 3        3       tr3   3 82.77806 20.10788</code></pre>
</div>
<div id="perform-three-dimensional-scan-as-first-step-in-pleiotropy-v-separate-qtl-hypothesis-test" class="section level1">
<h1>Perform three-dimensional scan as first step in pleiotropy v separate QTL hypothesis test</h1>
<p>We now have the inputs that we need to do a pleiotropy vs. separate QTL test. We have the founder allele dosages for one chromosome, <em>i.e.</em>, Chr 3, in the R object <code>pp</code>, the matrix of two trait measurements in <code>Y</code>, and a LOCO-derived kinship matrix. We also specify, via the <code>start_snp</code> argument, the starting point for the two-dimensional scan within the array of founder allele dosages. Here, we choose the 38th marker in the array as the starting point. Via the <code>n_snp</code> argument, we specify the number of markers to include in the two-dimensional scan. Here, we input 15, so that we fit the trivariate linear mixed effects model at 15<em>15</em>15 = 3375 ordered triples of markers. In practice, we usually use between 100 and 300 markers for most bivariate scans.</p>
<p>Lastly, we specify the number of cores to use, with the <code>n_cores</code> argument. We set it to 1 here, to ensure that the vignette can be run by CRAN. However, in practice, you may wish to increase the number of cores to accelerate computing.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>start_index &lt;-<span class="st"> </span><span class="dv">43</span></span>
<span id="cb17-2"><a href="#cb17-2"></a>out &lt;-<span class="st"> </span><span class="kw">scan_pvl</span>(<span class="dt">probs =</span> pp,</span>
<span id="cb17-3"><a href="#cb17-3"></a>                <span class="dt">pheno =</span> Y,</span>
<span id="cb17-4"><a href="#cb17-4"></a>                <span class="dt">kinship =</span> kinship<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>,</span>
<span id="cb17-5"><a href="#cb17-5"></a>                <span class="dt">start_snp =</span> start_index,</span>
<span id="cb17-6"><a href="#cb17-6"></a>                <span class="dt">n_snp =</span> <span class="dv">15</span></span>
<span id="cb17-7"><a href="#cb17-7"></a>                )</span></code></pre></div>
<pre><code>## starting covariance matrices estimation with data from 261 subjects.</code></pre>
<pre><code>## covariance matrices estimation completed.</code></pre>
<div id="create-a-profile-lod-plot-to-visualize-results-of-three-dimensional-scan" class="section level4">
<h4>Create a profile LOD plot to visualize results of three-dimensional scan</h4>
<p>To visualize results from our two-dimensional scan, we calculate profile LOD for each trait. The code below makes use of the R package <code>ggplot2</code> to plot profile LODs over the scan region.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>out</span></code></pre></div>
<pre><code>## # A tibble: 3,375 x 4
##    Var1               Var2               Var3               log10lik
##    &lt;chr&gt;              &lt;chr&gt;              &lt;chr&gt;                 &lt;dbl&gt;
##  1 backupUNC030474070 backupUNC030474070 backupUNC030474070    -559.
##  2 backupUNC030474244 backupUNC030474070 backupUNC030474070    -559.
##  3 UNC030103315       backupUNC030474070 backupUNC030474070    -559.
##  4 UNC030107226       backupUNC030474070 backupUNC030474070    -559.
##  5 JAX00527615        backupUNC030474070 backupUNC030474070    -558.
##  6 UNC030484421       backupUNC030474070 backupUNC030474070    -557.
##  7 JAX00109598        backupUNC030474070 backupUNC030474070    -557.
##  8 backupUNC030619800 backupUNC030474070 backupUNC030474070    -558.
##  9 UNC030314351       backupUNC030474070 backupUNC030474070    -559.
## 10 backupUNC030491543 backupUNC030474070 backupUNC030474070    -559.
## # … with 3,365 more rows</code></pre>
<p>We see that <code>out</code> is a tibble, as expected. The first three columns contain the marker ids for each ordered triple of markers. The last column contains the log-likelihood values.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="kw">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb26-2"><a href="#cb26-2"></a>out_lods &lt;-<span class="st"> </span>out <span class="op">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="st">  </span><span class="kw">calc_profile_lods</span>() <span class="op">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4"></a><span class="st">  </span><span class="kw">add_pmap</span>(<span class="dt">pmap =</span> DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>)</span></code></pre></div>
<pre><code>## Joining, by = &quot;marker&quot;</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>out_lods <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-2"><a href="#cb28-2"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> marker_position, <span class="dt">y =</span> profile_lod, <span class="dt">colour =</span> trait))</span></code></pre></div>
<p><img src="d-variate-analysis_files/figure-html/profile-plot-1.png" width="700" /></p>
</div>
<div id="calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="section level4">
<h4>Calculate the likelihood ratio test statistic for pleiotropy v separate QTL</h4>
<p>We use the function <code>calc_lrt_tib</code> to calculate the likelihood ratio test statistic value for the specified traits and specified genomic region.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>(lrt &lt;-<span class="st"> </span><span class="kw">max</span>(out_lods<span class="op">$</span>profile_lod))</span></code></pre></div>
<pre><code>## [1] 1.012118</code></pre>
</div>
<div id="bootstrap-analysis-to-get-p-values" class="section level3">
<h3>Bootstrap analysis to get p-values</h3>
<p>The calibration of test statistic values to get p-values uses bootstrap methods because we don’t know the theoretical distribution of the test statistic under the null hypothesis. Thus, we use a bootstrap approach to obtain an empirical distribution of test statistic values under the null hypothesis of the presence of one pleiotropic locus.</p>
<p>We will use the function <code>boot_pvl</code> from our package <code>qtl2pleio</code>.</p>
<p>We use a parametric bootstrap strategy in which we first use the studied phenotypes to infer the values of model parameters. Once we have the inferred values of the model parameters, we simulate phenotypes from the pleiotropy model (with the inferred parameter values).</p>
<p>A natural question that arises is “which marker’s allele probabilities do we use when simulating phenotypes?” We use the marker that, under the null hypothesis, <em>i.e.</em>, under the pleiotropy constraint, yields the greatest value of the log-likelihood.</p>
<p>Before we call <code>boot_pvl</code>, we need to identify the index (on the chromosome under study) of the marker that maximizes the likelihood under the pleiotropy constraint. To do this, we use the <code>qtl2pleio</code> function <code>find_pleio_peak_tib</code>.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>(pleio_index &lt;-<span class="st"> </span><span class="kw">find_pleio_peak_tib</span>(out, <span class="dt">start_snp =</span> start_index))</span></code></pre></div>
<pre><code>## log10lik8 
##        50</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="kw">set.seed</span>(<span class="dv">2018</span> <span class="op">-</span><span class="st"> </span><span class="dv">11</span> <span class="op">-</span><span class="st"> </span><span class="dv">25</span>)</span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw">suppressMessages</span>(b_out &lt;-<span class="st"> </span><span class="kw">boot_pvl</span>(<span class="dt">probs =</span> pp,</span>
<span id="cb33-3"><a href="#cb33-3"></a>                                   <span class="dt">pheno =</span> Y,</span>
<span id="cb33-4"><a href="#cb33-4"></a>                                   <span class="dt">pleio_peak_index =</span> pleio_index,</span>
<span id="cb33-5"><a href="#cb33-5"></a>                                   <span class="dt">kinship =</span> kinship<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, </span>
<span id="cb33-6"><a href="#cb33-6"></a>                                   <span class="dt">nboot =</span> <span class="dv">2</span>,</span>
<span id="cb33-7"><a href="#cb33-7"></a>                                   <span class="dt">start_snp =</span> start_index,</span>
<span id="cb33-8"><a href="#cb33-8"></a>                                   <span class="dt">n_snp =</span> <span class="dv">15</span></span>
<span id="cb33-9"><a href="#cb33-9"></a>                                   )</span>
<span id="cb33-10"><a href="#cb33-10"></a>                 )</span></code></pre></div>
<p>The argument <code>nboot</code> indicates the number of bootstrap samples that will be created and analyzed. Here, we set <code>nboot = 2</code>, so we expect to see returned a numeric vector of length 2, where each entry is a LRT statistic value from a distinct bootstrap sample.</p>
<p>Finally, we determine a bootstrap p-value in the usual method. We treat the bootstrap samples’ test statistics as an empirical distribution of the test statistic under the null hypothesis of pleiotropy. Thus, to get a p-value, we want to ask “What is the probability, under the null hypothesis, of observing a test statistic value that is at least as extreme as that which we observed?”</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>b_out</span></code></pre></div>
<pre><code>## [1] 0.0000000 0.2445236</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>(pvalue &lt;-<span class="st"> </span><span class="kw">mean</span>(b_out <span class="op">&gt;=</span><span class="st"> </span>lrt))</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>In practice, one would want to use many more bootstrap samples to achieve an empirical distribution that is closer to the theoretical distribution of the test statistic under the null hypothesis.</p>
<p>However, if one wants to perform analyses with a reasonable number - say 400 - bootstrap samples, this will take a very long time - many days - on a single laptop computer. We have used a series of computer clusters that are coordinated by the University of Wisconsin-Madison’s Center for High-throughput Computing (<a href="http://chtc.cs.wisc.edu" class="uri">http://chtc.cs.wisc.edu</a>). We typically are able to analyze 1000 bootstrap samples - for a bivariate analysis - in less than 24 hours with this service.</p>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.2 (2020-06-22)
##  os       Ubuntu 20.04 LTS            
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       America/New_York            
##  date     2020-07-08                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package     * version    date       lib source        
##  assertthat    0.2.1      2019-03-21 [2] CRAN (R 4.0.0)
##  backports     1.1.8      2020-06-17 [2] CRAN (R 4.0.0)
##  bit           1.1-15.2   2020-02-10 [2] CRAN (R 4.0.0)
##  bit64         0.9-7      2017-05-08 [2] CRAN (R 4.0.0)
##  blob          1.2.1      2020-01-20 [2] CRAN (R 4.0.0)
##  callr         3.4.3      2020-03-28 [2] CRAN (R 4.0.0)
##  cli           2.0.2      2020-02-28 [2] CRAN (R 4.0.0)
##  codetools     0.2-16     2018-12-24 [2] CRAN (R 4.0.0)
##  colorspace    1.4-1      2019-03-18 [2] CRAN (R 4.0.0)
##  crayon        1.3.4      2017-09-16 [2] CRAN (R 4.0.0)
##  data.table    1.12.8     2019-12-09 [2] CRAN (R 4.0.0)
##  DBI           1.1.0      2019-12-15 [2] CRAN (R 4.0.0)
##  desc          1.2.0      2018-05-01 [2] CRAN (R 4.0.0)
##  devtools      2.3.0      2020-04-10 [2] CRAN (R 4.0.0)
##  digest        0.6.25     2020-02-23 [2] CRAN (R 4.0.0)
##  dplyr       * 1.0.0      2020-05-29 [2] CRAN (R 4.0.0)
##  ellipsis      0.3.1      2020-05-15 [2] CRAN (R 4.0.0)
##  evaluate      0.14       2019-05-28 [2] CRAN (R 4.0.0)
##  fansi         0.4.1      2020-01-08 [2] CRAN (R 4.0.0)
##  farver        2.0.3      2020-01-16 [2] CRAN (R 4.0.0)
##  fs            1.4.2      2020-06-30 [2] CRAN (R 4.0.0)
##  furrr         0.1.0      2018-05-16 [2] CRAN (R 4.0.0)
##  future        1.17.0     2020-04-18 [2] CRAN (R 4.0.0)
##  gemma2        0.1.1.9000 2020-07-04 [2] local         
##  generics      0.0.2      2018-11-29 [2] CRAN (R 4.0.0)
##  ggplot2     * 3.3.2      2020-06-19 [2] CRAN (R 4.0.0)
##  globals       0.12.5     2019-12-07 [2] CRAN (R 4.0.0)
##  glue          1.4.1      2020-05-13 [2] CRAN (R 4.0.0)
##  gtable        0.3.0      2019-03-25 [2] CRAN (R 4.0.0)
##  htmltools     0.5.0      2020-06-16 [2] CRAN (R 4.0.0)
##  jsonlite      1.7.0      2020-06-25 [2] CRAN (R 4.0.0)
##  knitr         1.29       2020-06-23 [2] CRAN (R 4.0.0)
##  labeling      0.3        2014-08-23 [2] CRAN (R 4.0.0)
##  lattice       0.20-41    2020-04-02 [2] CRAN (R 4.0.0)
##  lifecycle     0.2.0      2020-03-06 [2] CRAN (R 4.0.0)
##  listenv       0.8.0      2019-12-05 [2] CRAN (R 4.0.0)
##  magrittr      1.5        2014-11-22 [2] CRAN (R 4.0.0)
##  MASS          7.3-51.6   2020-04-26 [2] CRAN (R 4.0.0)
##  Matrix        1.2-18     2019-11-27 [2] CRAN (R 4.0.0)
##  memoise       1.1.0      2017-04-21 [2] CRAN (R 4.0.0)
##  munsell       0.5.0      2018-06-12 [2] CRAN (R 4.0.0)
##  pillar        1.4.4      2020-05-05 [2] CRAN (R 4.0.0)
##  pkgbuild      1.0.8      2020-05-07 [2] CRAN (R 4.0.0)
##  pkgconfig     2.0.3      2019-09-22 [2] CRAN (R 4.0.0)
##  pkgdown       1.5.1      2020-04-09 [2] CRAN (R 4.0.0)
##  pkgload       1.1.0      2020-05-29 [2] CRAN (R 4.0.0)
##  prettyunits   1.1.1      2020-01-24 [2] CRAN (R 4.0.0)
##  processx      3.4.2      2020-02-09 [2] CRAN (R 4.0.0)
##  ps            1.3.3      2020-05-08 [2] CRAN (R 4.0.0)
##  purrr       * 0.3.4      2020-04-17 [2] CRAN (R 4.0.0)
##  qtl2        * 0.22-8     2020-06-26 [2] CRAN (R 4.0.0)
##  qtl2pleio   * 1.3.0      2020-07-08 [1] local         
##  R6            2.4.1      2019-11-12 [2] CRAN (R 4.0.0)
##  Rcpp          1.0.4.6    2020-04-09 [2] CRAN (R 4.0.0)
##  remotes       2.1.1      2020-02-15 [2] CRAN (R 4.0.0)
##  rlang         0.4.6      2020-05-02 [2] CRAN (R 4.0.0)
##  rmarkdown     2.3        2020-06-18 [2] CRAN (R 4.0.0)
##  rprojroot     1.3-2      2018-01-03 [2] CRAN (R 4.0.0)
##  RSQLite       2.2.0      2020-01-07 [2] CRAN (R 4.0.0)
##  rstudioapi    0.11       2020-02-07 [2] CRAN (R 4.0.0)
##  scales        1.1.1      2020-05-11 [2] CRAN (R 4.0.0)
##  sessioninfo   1.1.1      2018-11-05 [2] CRAN (R 4.0.0)
##  stringi       1.4.6      2020-02-17 [2] CRAN (R 4.0.0)
##  stringr       1.4.0      2019-02-10 [2] CRAN (R 4.0.0)
##  testthat      2.3.2      2020-03-02 [2] CRAN (R 4.0.0)
##  tibble        3.0.1      2020-04-20 [2] CRAN (R 4.0.0)
##  tidyselect    1.1.0      2020-05-11 [2] CRAN (R 4.0.0)
##  usethis       1.6.1      2020-04-29 [2] CRAN (R 4.0.0)
##  utf8          1.1.4      2018-05-24 [2] CRAN (R 4.0.0)
##  vctrs         0.3.1      2020-06-05 [2] CRAN (R 4.0.0)
##  withr         2.2.0      2020-04-20 [2] CRAN (R 4.0.0)
##  xfun          0.15       2020-06-21 [2] CRAN (R 4.0.0)
##  yaml          2.2.1      2020-02-01 [2] CRAN (R 4.0.0)
## 
## [1] /tmp/RtmpBdHTVG/temp_libpath1d374c23fb8d
## [2] /home/fred/R/x86_64-pc-linux-gnu-library/4.0
## [3] /usr/local/lib/R/site-library
## [4] /usr/lib/R/site-library
## [5] /usr/lib/R/library</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc">
      <h2 data-toc-skip>Contents</h2>
    </nav>
      </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by Frederick J Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>

