<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ qtl2pleio</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-46757486-5"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-46757486-5');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/d-variate-analysis.html">Multivariate analysis for two or more traits</a>
    </li>
    <li>
      <a href="../articles/multivariate-one-QTL-scans.html">Multivariate, one-QTL scans</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fboehm/qtl2pleio/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="add_pmap.html">add_pmap()</a></code> </p>
        </td>
        <td><p>Add physical map contents to tibble</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="boot_pvl.html">boot_pvl()</a></code> </p>
        </td>
        <td><p>Perform bootstrap sampling and calculate test statistic for each bootstrap sample</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_Bhat.html">calc_Bhat()</a></code> </p>
        </td>
        <td><p>Calculate estimated allele effects, B matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_Sigma.html">calc_Sigma()</a></code> </p>
        </td>
        <td><p>Calculate the phenotypes covariance matrix Sigma</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_covs.html">calc_covs()</a></code> </p>
        </td>
        <td><p>Calculate Vg and Ve from d-variate phenotype and kinship</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_invsqrt_mat.html">calc_invsqrt_mat()</a></code> </p>
        </td>
        <td><p>Calculate matrix inverse square root for a covariance matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_lrt_tib.html">calc_lrt_tib()</a></code> </p>
        </td>
        <td><p>Calculate a likelihood ratio test statistic from the output of scan_pvl()</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_profile_lods.html">calc_profile_lods()</a></code> </p>
        </td>
        <td><p>Calculate profile lods for all traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc_sqrt_mat.html">calc_sqrt_mat()</a></code> </p>
        </td>
        <td><p>Calculate matrix square root for a covariance matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check_identical.html">check_identical()</a></code> </p>
        </td>
        <td><p>Check whether a vector, x, has all its entries equal to its first entry</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check_missingness.html">check_missingness()</a></code> </p>
        </td>
        <td><p>Check for missingness in phenotypes or covariates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_to_scan1_output.html">convert_to_scan1_output()</a></code> </p>
        </td>
        <td><p>Convert `scan_multi_oneqtl` output of `qtl2::scan1` output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="find_pleio_peak_tib.html">find_pleio_peak_tib()</a></code> </p>
        </td>
        <td><p>Find the marker index corresponding to the peak of the pleiotropy trace in a tibble where the last column contains log likelihood values and the first d columns contain marker ids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fit1_pvl.html">fit1_pvl()</a></code> </p>
        </td>
        <td><p>Fit a model for a specified d-tuple of markers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_effects.html">get_effects()</a></code> </p>
        </td>
        <td><p>Extract founder allele effects at a single marker from output of qtl2::scan1coef</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_id2keep.html">make_id2keep()</a></code> </p>
        </td>
        <td><p>Identify shared subject ids among all inputs: covariates, allele probabilities array, kinship, and phenotypes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_pvl.html">plot_pvl()</a></code> </p>
        </td>
        <td><p>Plot tidied results of a pvl scan</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prep_X_list.html">prep_X_list()</a></code> </p>
        </td>
        <td><p>Create a list of component X matrices for input to stagger_mats, to ultimately create design matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prep_mytab.html">prep_mytab()</a></code> </p>
        </td>
        <td><p>Prepare mytab object for use within scan_pvl R code</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="process_inputs.html">process_inputs()</a></code> </p>
        </td>
        <td><p>Process inputs to scan functions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="qtl2pleio.html">qtl2pleio</a></code> </p>
        </td>
        <td><p>qtl2pleio.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rcpp_calc_Bhat.html">rcpp_calc_Bhat()</a></code> </p>
        </td>
        <td><p>Estimate allele effects matrix, B hat, with Rcpp functions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rcpp_calc_Bhat2.html">rcpp_calc_Bhat2()</a></code> </p>
        </td>
        <td><p>Estimate allele effects matrix, B hat, with Rcpp functions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rcpp_log_dmvnorm2.html">rcpp_log_dmvnorm2()</a></code> </p>
        </td>
        <td><p>Calculate log likelihood for a multivariate normal</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scan_multi_onechr.html">scan_multi_onechr()</a></code> </p>
        </td>
        <td><p>Perform multivariate, one-QTL model fitting for markers on one chromosome</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scan_multi_oneqtl.html">scan_multi_oneqtl()</a></code> </p>
        </td>
        <td><p>Perform multivariate, one-QTL model fitting for markers on all chromosomes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scan_multi_oneqtl_perm.html">scan_multi_oneqtl_perm()</a></code> </p>
        </td>
        <td><p>Permute the phenotypes matrix and then scan the genome. Record the genomewide greatest LOD score for each permuted data set.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scan_pvl.html">scan_pvl()</a></code> </p>
        </td>
        <td><p>Perform model fitting for all ordered pairs of markers in a genomic region of interest</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sim1.html">sim1()</a></code> </p>
        </td>
        <td><p>Simulate a single multivariate data set consisting of n subjects and d phenotypes for each</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset_input.html">subset_input()</a></code> </p>
        </td>
        <td><p>Subset an input object - allele probabilities array or phenotypes matrix or covariates matrix. Kinship has its own subset function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset_kinship.html">subset_kinship()</a></code> </p>
        </td>
        <td><p>Subset a kinship matrix to include only those subjects present in all inputs</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Frederick J Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


